// @framer/godaddy.maui/code/Foundation/List.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"
import {
    themeColors,
    textStyles,
    glyphOptions,
    glyphOptionTitles,
} from "./Shared"
import { Text } from "./Text"
import { Glyph } from "./Glyph"
import { Divider } from "./Divider"

// const bounds = "hsla(0, 100%, 50%, 0.5)"
const bounds = "none"

/* ----------------------------------------------------------------------------------------------------- */
/*                                             COMPONENT                                                 */
/* ----------------------------------------------------------------------------------------------------- */

export function List(props) {
    console.log(textStyles.body)

    /* ------------------------------------------------------------------------------------------------- */
    /*                                          CONTENT                                                  */
    /* ------------------------------------------------------------------------------------------------- */
    const getListContent = () => {
        switch (props.type) {
            case "label":
                return (
                    <Frame size="100%" background={bounds}>
                        <Text
                            widthType={"auto"}
                            left={24}
                            right={24}
                            center="y"
                            content={`*${props.label}*`}
                            style={"body"}
                            color={"contentPrimary"}
                            align={"leftAlign"}
                            background={bounds}
                            mode={props.mode}
                        />
                        {props.divider === "show" && (
                            <Divider left={24} right={24} />
                        )}
                    </Frame>
                )
                break
            case "website":
                return (
                    <Frame width="100%" height={108}>
                        ad
                    </Frame>
                )
                break
        }
    }
    const listContent = React.useMemo(() => getListContent(), [props])

    /* ------------------------------------------------------------------------------------------------- */
    /*                                           MAIN                                                    */
    /* ------------------------------------------------------------------------------------------------- */
    return (
        <Frame size="100%" background={bounds}>
            {listContent}
        </Frame>
    )
}

/* ----------------------------------------------------------------------------------------------------- */
/*                                           PROPERTIES                                                  */
/* ----------------------------------------------------------------------------------------------------- */

List.defaultProps = {
    width: 375,
    height: 54,
    type: "label",
    label: "Placeholder Text",
    divider: "show",
    mode: "light",
}

addPropertyControls(List, {
    type: {
        title: "Type",
        type: ControlType.Enum,
        options: ["label", "labelGlyph", "website"],
        optionTitles: ["Label", "Label ‹–› Glyph", "Website"],
    },
    label: {
        title: "Label",
        type: ControlType.String,
    },
    divider: {
        title: "Divider",
        type: ControlType.SegmentedEnum,
        options: ["hide", "show"],
        optionTitles: ["Hide", "Show"],
    },
    mode: {
        title: "Mode",
        type: ControlType.SegmentedEnum,
        options: ["light", "dark"],
        optionTitles: ["Light", "Dark"],
    },
})
