// @framer/godaddy.maui/code/Foundation/Shared.tsx
import * as React from "react"
import { Frame } from "framer"
import { colors } from "../canvas"

//  ------------------------------------------------------------------------------------
//  COLORS

export const themeColors = {
    light: {
        // Content (Text and Glyphs)
        contentPrimary: colors.LM_content_primary,
        contentSecondary: colors.LM_content_secondary,
        contentTertiary: colors.LM_content_tertiary,
        contentHighlight: colors.LM_content_highlight,
        contentImportant: colors.LM_content_important,
        contentButtonPrimary: colors.LM_content_btn_primary,
        contentButtonSecondary: colors.LM_content_btn_secondary,

        // Backgrounds (Surfaces)
        bgBase: colors.LM_bg_base,
        bgNavIOS: colors.LM_bg_nav_ios,
        bgNavAndroid: colors.LM_bg_nav_android,
        bgSheetIOS: colors.LM_bg_sheet_ios,
        bgSheetAndroid: colors.LM_bg_sheet_android,
        bgOverlay: colors.LM_bg_overlay,
        bgDivider: colors.LM_bg_divider,
        bgPlaceholder: colors.LM_bg_placeholder,

        // Backgrounds (Buttons, textfields, etc)
        bgPrimary: colors.LM_bg_primary,
        bgSecondary: colors.LM_bg_secondary,
        bgTertiary: colors.LM_bg_tertiary,
        bgHighlight: colors.LM_bg_highlight,
        bgImportant: colors.LM_bg_important,

        // Backgrounds (Paint and decorations)
        bgTeal: colors.LM_bg_teal,
        bgBlue: colors.LM_bg_blue,
        bgBrown: colors.LM_bg_brown,
        bgGreen: colors.LM_bg_green,
        bgMagenta: colors.LM_bg_magenta,
        bgOrange: colors.LM_bg_orange,
        bgPurple: colors.LM_bg_purple,
        bgYellow: colors.LM_bg_yellow,
        bgSilver: colors.LM_bg_silver,
    },
    dark: {
        // Content (Text and Glyphs)
        contentPrimary: colors.DM_content_primary,
        contentSecondary: colors.DM_content_secondary,
        contentTertiary: colors.DM_content_tertiary,
        contentHighlight: colors.DM_content_highlight,
        contentImportant: colors.DM_content_important,
        contentButtonPrimary: colors.DM_content_btn_primary,
        contentButtonSecondary: colors.DM_content_btn_secondary,

        // Backgrounds (Surfaces)
        bgBase: colors.DM_bg_base,
        bgNavIOS: colors.DM_bg_nav_ios,
        bgNavAndroid: colors.DM_bg_nav_android,
        bgSheetIOS: colors.DM_bg_sheet_ios,
        bgSheetAndroid: colors.DM_bg_sheet_android,
        bgOverlay: colors.DM_bg_overlay,
        bgDivider: colors.DM_bg_divider,
        bgPlaceholder: colors.DM_bg_placeholder,

        // Backgrounds (Buttons, textfields, etc)
        bgPrimary: colors.DM_bg_primary,
        bgSecondary: colors.DM_bg_secondary,
        bgTertiary: colors.DM_bg_tertiary,
        bgHighlight: colors.DM_bg_highlight,
        bgImportant: colors.DM_bg_important,

        // Backgrounds (Paint and decorations)
        bgTeal: colors.DM_bg_teal,
        bgBlue: colors.DM_bg_blue,
        bgBrown: colors.DM_bg_brown,
        bgGreen: colors.DM_bg_green,
        bgMagenta: colors.DM_bg_magenta,
        bgOrange: colors.DM_bg_orange,
        bgPurple: colors.DM_bg_purple,
        bgYellow: colors.DM_bg_yellow,
        bgSilver: colors.DM_bg_silver,
    },
}

//  ------------------------------------------------------------------------------------
//  TEXT STYLES

export const textStyles = {
    large: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "40px",
        lineHeight: "48px",
    },
    h1: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "30px",
        lineHeight: "36px",
    },
    h2: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "26px",
        lineHeight: "32px",
    },
    h3: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "22px",
        lineHeight: "28px",
    },
    h4: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "20px",
        lineHeight: "26px",
    },
    title: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "18px",
        lineHeight: "24px",
    },
    body: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "16px",
        lineHeight: "22px",
    },
    detail: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "14px",
        lineHeight: "20px",
    },
    caption1: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "13px",
        lineHeight: "18px",
    },
    caption2: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "12px",
        lineHeight: "16px",
    },
    small: {
        fontFamily: "GDSherpa-Semibold",
        fontSize: "10px",
        lineHeight: "14px",
    },
}

//  ------------------------------------------------------------------------------------
//  GLYPHS

export const glyphOptions = [
    "arrowLeft",
    "arrowRight",
    "arrowUp",
    "arrowDown",
    "plus",
    "plus12",
    "minus",
    "minus12",
    "plusSquare",
    "minusSquare",
    "check",
    "x",
    "cartOpen",
    "cartClose",
    "eye",
    "search",
    "search20",
    "connect",
    "disconnect",
    "renew",
    "siteSwitcher",
    "siteManager",
    "website",
    "domain",
    "person",
    "personCircle",
    "personCircleFill",
    "bell",
    "play",
    "more",
    "calendar",
    "shoppingBag",
    "penPaper",
    "penLine",
    "photo",
    "photoSquare",
    "social",
    "star",
    "conversations",
    "numpad",
    "contacts",
    "mention",
    "shareIOS",
    "shareAndroid",
    "speaker",
    "slashCircle",
    "message",
    "messageFill",
    "phone",
    "phoneFill",
    "voicemail",
    "cog",
    "trashCan",
    "coupon",
    "caretDown12",
    "clear20",
]

export const glyphOptionTitles = [
    "Arrow - Left",
    "Arrow - Right",
    "Arrow - Up",
    "Arrow - Down",
    "Plus",
    "Plus (12)",
    "Minus",
    "Minus (12)",
    "Plus - Square",
    "Minus - Square",
    "Check",
    "X",
    "Cart - Open",
    "Cart - Close",
    "Eye",
    "Search",
    "Search (20)",
    "Connect",
    "Disconnect",
    "Renew",
    "Site Switcher",
    "Site Manager",
    "Website",
    "Domain",
    "Person",
    "Person - Circle",
    "Person - Circle - Fill",
    "Bell",
    "Play",
    "More",
    "Calendar",
    "Shopping Bag",
    "Pen & Paper",
    "Pen & Line",
    "Photo",
    "Photo - Square",
    "Social",
    "Star",
    "Conversations",
    "Numpad",
    "Contacts",
    "Mention",
    "Share - iOS",
    "Share - Android",
    "Speaker",
    "Slash - Circle",
    "Message",
    "Message - Fill",
    "Phone",
    "PhoneFill",
    "Voicemail",
    "Cog",
    "Trash Can",
    "Coupon",
    "Caret Down (12)",
    "Clear (20)",
]

export const glyphOptions12 = ["plus12", "minus12", "caretDown12"]

export const glyphOptionTitles12 = ["Plus", "Minus", "Caret Down"]

//  ------------------------------------------------------------------------------------
//  TRANSITIONS

export const transitions = {
    // Used in: buttons
    bounceSoft: {
        type: "spring",
        stiffness: 150,
        damping: 15,
        mass: 2,
        velocity: 0,
    },
    bounceStrong: {
        type: "spring",
        stiffness: 160,
        damping: 8,
        mass: 1,
        velocity: 1.2,
    },
    // Ued in: buttons
    fadeSlow: {
        type: "tween",
        duration: 0.6,
    },
    fadeFast: {
        type: "tween",
        duration: 0.15,
    },
    // Used in: App Open
    expand: {
        type: "spring",
        damping: 80,
        mass: 1,
        stiffness: 240,
        velocity: 400,
    },
    // testing...
    push: {
        type: "spring",
        stiffness: 80,
        damping: 10,
        mass: 0.3,
    },
    fade: {
        type: "tween",
        duration: 0.3,
    },
    slide: {
        ease: [0.08, 0.82, 0.54, 0.95],
        duration: 0.8,
    },
    bezier: {
        ease: [0.825, -0.38, 0.275, 0.99],
        duration: 0.8,
    },
}

//  ------------------------------------------------------------------
//  DELAY

export const delay = milliseconds => {
    return new Promise(resolve => setTimeout(resolve, milliseconds * 1000))
}
