// @framer/godaddy.maui/code/Foundation/Cursor.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, useAnimation } from "framer"
// @ts-ignore
import { themeColors } from "./Shared"

export function Cursor(props) {
    return (
        <Frame
            width={2}
            height={props.height}
            radius={100}
            background={themeColors[props.mode]["contentHighlight"]}
            style={{ display: "inline" }}
            visible={props.visible}
            animate={
                props.state === true ? { opacity: [0, 0, 1, 1, 0, 0] } : {}
            }
            transition={
                props.state === true
                    ? {
                          duration: 1,
                          times: [0, 0.2, 0.3, 0.6, 0.7, 1],
                          loop: Infinity,
                      }
                    : {}
            }
        />
    )
}

Cursor.defaultProps = {
    width: 2,
    height: 26,
    state: false,
    mode: "light",
}

//  ------------------------------------------------------------------------------------
//  PROPERTY CONTROLS

addPropertyControls(Cursor, {
    mode: {
        title: "Mode",
        type: ControlType.SegmentedEnum,
        options: ["light", "dark"],
        optionTitles: ["Light", "Dark"],
    },
})
