// @framer/godaddy.maui/code/Foundation/Divider.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"
import { themeColors } from "./Shared"

export function Divider(props) {
    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                   COLOR                                                       */
    /* ------------------------------------------------------------------------------------------------------------- */

    const getDividerColor = () => {
        switch (props.mode) {
            case "light":
                return themeColors.light.bgDivider
                break
            case "dark":
                return themeColors.dark.bgDivider
                break
        }
    }

    const dividerColor = React.useMemo(() => getDividerColor(), [props])

    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                   STYLE                                                       */
    /* ------------------------------------------------------------------------------------------------------------- */

    const getDividerStyle = () => {
        switch (props.type) {
            case "top":
                return (
                    <Frame
                        height="100%"
                        left={props.left}
                        right={props.right}
                        background="none"
                        style={{
                            borderTopWidth: 1 / devicePixelRatio,
                            borderTopStyle: "solid",
                            borderColor: dividerColor,
                        }}
                    />
                )
                break
            case "bottom":
                return (
                    <Frame
                        height="100%"
                        left={props.left}
                        right={props.right}
                        background="none"
                        style={{
                            borderBottomWidth: 1 / devicePixelRatio,
                            borderBottomStyle: "solid",
                            borderColor: dividerColor,
                        }}
                    />
                )
                break
        }
    }

    const dividerContent = React.useMemo(() => getDividerStyle(), [props])

    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                      MAIN                                                     */
    /* ------------------------------------------------------------------------------------------------------------- */

    return (
        <Frame
            name={"divider"}
            width={"100%"}
            height={1}
            bottom={0}
            background={"none"}
        >
            {dividerContent}
        </Frame>
    )
}

/* ------------------------------------------------------------------------------------------------------------- */
/*                                                   PROPERTIES                                                  */
/* ------------------------------------------------------------------------------------------------------------- */

Divider.defaultProps = {
    height: 1,
    left: 0,
    right: 0,
    type: "bottom",
    mode: "light",
}

addPropertyControls(Divider, {
    type: {
        title: "Type",
        type: ControlType.SegmentedEnum,
        options: ["top", "bottom"],
        optionTitles: ["Top", "Bottom"],
    },
    mode: {
        title: "Mode",
        type: ControlType.SegmentedEnum,
        options: ["light", "dark"],
        optionTitles: ["Light", "Dark"],
    },
})
