// @framer/godaddy.maui/code/Foundation/Glyph.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"
// @ts-ignore
import { themeColors, glyphOptions, glyphOptionTitles } from "./Shared"

export function Glyph(props) {
    const { glyph, color, weight, mode, ...rest } = props

    //  ------------------------------------------------------------------
    //  RENDER ICON

    const getSVG = () => {
        return icons[props.glyph](
            props.mode === "light"
                ? themeColors.light[props.color]
                : themeColors.dark[props.color],
            props.weight === "default"
                ? 2
                : props.weight === "medium"
                ? 1.67
                : 1.33
        )
    }

    const glyphContent = React.useMemo(() => getSVG(), [
        glyph,
        color,
        weight,
        mode,
    ])

    //  ------------------------------------------------------------------
    //  MAIN

    return <Frame {...rest}>{glyphContent}</Frame>
}

//  ------------------------------------------------------------------
//  PROPS

Glyph.defaultProps = {
    width: 24,
    height: 24,
    background: "none",
    glyph: "calendar",
    color: "contentPrimary",
    weight: "default",
    mode: "light",
}

addPropertyControls(Glyph, {
    glyph: {
        type: ControlType.Enum,
        title: "Glyph",
        options: glyphOptions.sort(),
        optionTitles: glyphOptionTitles.sort(),
    },
    color: {
        title: "Color",
        type: ControlType.Enum,
        options: [
            "contentPrimary",
            "contentSecondary",
            "contentTertiary",
            "contentHighlight",
            "contentImportant",
            "contentButtonPrimary",
            "contentButtonSecondary",
        ],
        optionTitles: [
            "Primary",
            "Secondary",
            "Tertiary",
            "Highlight",
            "Important",
            "Button (Primary)",
            "Button (Secondary)",
        ],
    },
    weight: {
        title: "Weight",
        type: ControlType.Enum,
        options: ["light", "medium", "default"],
        optionTitles: ["Light", "Medium", "Default"],
    },
    mode: {
        type: ControlType.SegmentedEnum,
        title: "Mode",
        options: ["light", "dark"],
        optionTitles: ["Light", "Dark"],
    },
})

//  ------------------------------------------------------------------
//  SVGS

const icons = {
    /* ------------------- ARROW LEFT ------------------- */

    arrowLeft: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 11.5 4.5 L 4 12 L 11.5 19.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 4 12 L 21 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- ARROW RIGHT ------------------- */

    arrowRight: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <g>
                    <path
                        d="M 12.5 4.5 L 20 12 L 12.5 19.5"
                        fill="transparent"
                        strokeWidth={weight}
                        stroke={color}
                        strokeLinejoin="round"
                    ></path>
                    <path
                        d="M 3 12 L 20 12"
                        fill="transparent"
                        strokeWidth={weight}
                        stroke={color}
                    ></path>
                </g>
            </svg>
        )
    },

    /* ------------------- ARROW UP ------------------- */

    arrowUp: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 12 21 L 12 4"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 4.5 11 L 12 3.5 L 19.5 11"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- ARROW DOWN ------------------- */

    arrowDown: (color: string, weight: number) => {
        return (
            <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24">
                <path
                    d="M 12 3 L 12 20"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 4.5 12.5 L 12 20 L 19.5 12.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- PLUS ------------------- */

    plus: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 12 3 L 12 21"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 3 12 L 21 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- PLUS (12) ------------------- */

    plus12: (color: string, weight: number) => {
        return (
            <svg width="12" height="12">
                <path
                    d="M 6 1 L 6 11"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 1 6 L 11 6"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- MINUS ------------------- */

    minus: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 3 12 L 21 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- MINUS ------------------- */

    minus12: (color: string, weight: number) => {
        return (
            <svg width="12" height="12">
                <path
                    d="M 1 6 L 11 6"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- PLUS - SQUARE ------------------- */

    plusSquare: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 2 5 C 2 3.343 3.343 2 5 2 L 19 2 C 20.657 2 22 3.343 22 5 L 22 19 C 22 20.657 20.657 22 19 22 L 5 22 C 3.343 22 2 20.657 2 19 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 12 6 L 12 18"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 6 12 L 18 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- MINUS - SQUARE ------------------- */

    minusSquare: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 2 5 C 2 3.343 3.343 2 5 2 L 19 2 C 20.657 2 22 3.343 22 5 L 22 19 C 22 20.657 20.657 22 19 22 L 5 22 C 3.343 22 2 20.657 2 19 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 6 12 L 18 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- CHECK ------------------- */

    check: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 20.5 5 L 9 16.5 L 3.5 11"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- CLOSE ------------------- */

    x: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 4.5 4.5 L 19.5 19.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 19.5 4.5 L 4.5 19.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- CART OPEN ------------------- */

    cartOpen: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 9.5 20 C 9.776 20 10 20.224 10 20.5 C 10 20.776 9.776 21 9.5 21 C 9.224 21 9 20.776 9 20.5 C 9 20.224 9.224 20 9.5 20 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 18.5 20 C 18.776 20 19 20.224 19 20.5 C 19 20.776 18.776 21 18.5 21 C 18.224 21 18 20.776 18 20.5 C 18 20.224 18.224 20 18.5 20 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 0 4 L 4.5 4 L 8.5 15 L 19.5 15 L 22.5 7"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- CART CLOSE ------------------- */

    cartClose: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 9.5 20 C 9.776 20 10 20.224 10 20.5 C 10 20.776 9.776 21 9.5 21 C 9.224 21 9 20.776 9 20.5 C 9 20.224 9.224 20 9.5 20 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 18.5 20 C 18.776 20 19 20.224 19 20.5 C 19 20.776 18.776 21 18.5 21 C 18.224 21 18 20.776 18 20.5 C 18 20.224 18.224 20 18.5 20 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 0 4 L 4.5 4 L 8.5 15 L 19.5 15 L 22.5 7 L 6 7"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- EYE ------------------- */

    eye: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 1 12 C 1 12 4 4 12 4 C 20 4 23 12 23 12 C 23 12 20 20 12 20 C 4 20 1 12 1 12 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 12 9 C 13.657 9 15 10.343 15 12 C 15 13.657 13.657 15 12 15 C 10.343 15 9 13.657 9 12 C 9 10.343 10.343 9 12 9 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- SEARCH ------------------- */

    search: (color: string, weight: number) => {
        return (
            <svg width="24" height="24">
                <path
                    d="M 16 16 L 20.5 20.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 10.5 3.5 C 14.366 3.5 17.5 6.634 17.5 10.5 C 17.5 14.366 14.366 17.5 10.5 17.5 C 6.634 17.5 3.5 14.366 3.5 10.5 C 3.5 6.634 6.634 3.5 10.5 3.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- SEARCH 20  ------------------- */

    search20: (color: string, weight: number) => {
        return (
            <svg width="20" height="20">
                <path
                    d="M 13 13 L 18 18"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 8.5 2.5 C 11.814 2.5 14.5 5.186 14.5 8.5 C 14.5 11.814 11.814 14.5 8.5 14.5 C 5.186 14.5 2.5 11.814 2.5 8.5 C 2.5 5.186 5.186 2.5 8.5 2.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- CONNECT  ------------------- */

    connect: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 0 12 L 8 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 16 12 L 24 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 12 8 C 14.209 8 16 9.791 16 12 C 16 14.209 14.209 16 12 16 C 9.791 16 8 14.209 8 12 C 8 9.791 9.791 8 12 8 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- DISCONNECT  ------------------- */

    disconnect: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 15 12 L 24 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 0 12 L 9 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 12 16 L 18 8"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="square"
                ></path>
                <path
                    d="M 6 16 L 12 8"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="square"
                ></path>
            </svg>
        )
    },

    /* ------------------- RENEW  ------------------- */

    renew: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 16 13.5 L 20.5 18 L 16 22.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 20 18 L 7 18 C 5.343 18 4 16.657 4 15 L 4 13"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 8 10.5 L 3.5 6 L 8 1.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 4 6 L 17 6 C 18.657 6 20 7.343 20 9 L 20 11"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- SITE SWITCHER  ------------------- */

    siteSwitcher: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 14 7 L 14 3.5 C 14 2.672 13.328 2 12.5 2 L 3.5 2 C 2.672 2 2 2.672 2 3.5 L 2 15.5 C 2 16.328 2.672 17 3.5 17 L 10 17"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 10 8.5 C 10 7.672 10.672 7 11.5 7 L 20.5 7 C 21.328 7 22 7.672 22 8.5 L 22 20.5 C 22 21.328 21.328 22 20.5 22 L 11.5 22 C 10.672 22 10 21.328 10 20.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- SITE MANAGER  ------------------- */

    siteManager: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 14 15.5 C 14 14.672 14.672 14 15.5 14 L 20.5 14 C 21.328 14 22 14.672 22 15.5 L 22 20.5 C 22 21.328 21.328 22 20.5 22 L 15.5 22 C 14.672 22 14 21.328 14 20.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 2 15.5 C 2 14.672 2.672 14 3.5 14 L 8.5 14 C 9.328 14 10 14.672 10 15.5 L 10 20.5 C 10 21.328 9.328 22 8.5 22 L 3.5 22 C 2.672 22 2 21.328 2 20.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 14 3.5 C 14 2.672 14.672 2 15.5 2 L 20.5 2 C 21.328 2 22 2.672 22 3.5 L 22 8.5 C 22 9.328 21.328 10 20.5 10 L 15.5 10 C 14.672 10 14 9.328 14 8.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 2 3.5 C 2 2.672 2.672 2 3.5 2 L 8.5 2 C 9.328 2 10 2.672 10 3.5 L 10 8.5 C 10 9.328 9.328 10 8.5 10 L 3.5 10 C 2.672 10 2 9.328 2 8.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- WEBSITE  ------------------- */

    website: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 2 5 C 2 3.343 3.343 2 5 2 L 19 2 C 20.657 2 22 3.343 22 5 L 22 19 C 22 20.657 20.657 22 19 22 L 5 22 C 3.343 22 2 20.657 2 19 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 9.5 5.5 L 18.5 5.5"
                    fill="transparent"
                    strokeWidth={weight - 0.67}
                    stroke={color}
                    strokeLinecap="round"
                ></path>
                <path
                    d="M 5.5 5.5 L 6.5 5.5"
                    fill="transparent"
                    strokeWidth={weight - 0.67}
                    stroke={color}
                    strokeLinecap="round"
                ></path>
                <path
                    d="M 2 9 L 22 9"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- DOMAIN  ------------------- */

    domain: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 12 1.5 C 17.799 1.5 22.5 6.201 22.5 12 C 22.5 17.799 17.799 22.5 12 22.5 C 6.201 22.5 1.5 17.799 1.5 12 C 1.5 6.201 6.201 1.5 12 1.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 12 1.5 C 14.209 1.5 16 6.201 16 12 C 16 17.799 14.209 22.5 12 22.5 C 9.791 22.5 8 17.799 8 12 C 8 6.201 9.791 1.5 12 1.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 2 12 L 22 12"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- PERSON  ------------------- */

    person: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 2 23 C 2 23 2 21.5 2 19.5 C 2 17.5 3.751 16 5.5 15.5 C 7.249 15 9.667 15 12 15 C 14.332 15 16.751 15 18.5 15.5 C 20.249 16 22 17.51 22 19.5 C 22 21.469 22 23 22 23"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 12 1.5 C 14.75 1.5 16 3.291 16 5.5 C 16 7.709 15.5 11 12 11 C 8.5 11 8 7.709 8 5.5 C 8 3.291 9.25 1.5 12 1.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- PERSON CIRCLE  ------------------- */

    personCircle: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 12 1.5 C 17.799 1.5 22.5 6.201 22.5 12 C 22.5 17.799 17.799 22.5 12 22.5 C 6.201 22.5 1.5 17.799 1.5 12 C 1.5 6.201 6.201 1.5 12 1.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 12 6 C 14.063 6 15 7.5 15 9 C 15 10.5 14.25 12.5 12 12.5 C 9.75 12.5 9 10.5 9 9 C 9 7.5 9.938 6 12 6 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 4 19 C 5.81 17.19 7 16 12 16 C 17 16 18.19 17.19 20 19"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- PERSON CIRCLE FILL ------------------- */

    personCircleFill: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 12 0.5 C 18.351 0.5 23.5 5.649 23.5 12 C 23.5 18.351 18.351 23.5 12 23.5 C 5.649 23.5 0.5 18.351 0.5 12 C 0.5 5.649 5.649 0.5 12 0.5 Z M 4.358 17.644 C 6.088 19.983 8.867 21.5 12 21.5 C 15.133 21.5 17.912 19.983 19.642 17.644 C 18.013 16.036 16.664 15 12 15 C 7.336 15 5.987 16.036 4.358 17.644 Z M 9 8.5 C 9 10 9.75 12 12 12 C 14.25 12 15 10 15 8.5 C 15 7 14.063 5.5 12 5.5 C 9.938 5.5 9 7 9 8.5 Z"
                    fill={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- BELL ------------------- */

    bell: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 15 19.5 C 15 20.328 14.664 21.078 14.121 21.621 C 13.578 22.164 12.828 22.5 12 22.5 C 11.172 22.5 10.422 22.164 9.879 21.621 C 9.336 21.078 9 20.328 9 19.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 21 19 C 21 19 21 16.955 21 16 C 21 14.33 18.5 16 18.5 10 C 18.5 4 16 1.5 12 1.5 C 8 1.5 5.5 4 5.5 10 C 5.5 15.5 3 14.33 3 16 C 3 17.024 3 19 3 19 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- PLAY ------------------- */

    play: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 12 1.5 C 17.799 1.5 22.5 6.201 22.5 12 C 22.5 17.799 17.799 22.5 12 22.5 C 6.201 22.5 1.5 17.799 1.5 12 C 1.5 6.201 6.201 1.5 12 1.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 10.5 9 L 15.5 12 L 10.5 15 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- MORE ------------------- */

    more: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 20 11 C 20.552 11 21 11.448 21 12 C 21 12.552 20.552 13 20 13 C 19.448 13 19 12.552 19 12 C 19 11.448 19.448 11 20 11 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 12 11 C 12.552 11 13 11.448 13 12 C 13 12.552 12.552 13 12 13 C 11.448 13 11 12.552 11 12 C 11 11.448 11.448 11 12 11 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 4 11 C 4.552 11 5 11.448 5 12 C 5 12.552 4.552 13 4 13 C 3.448 13 3 12.552 3 12 C 3 11.448 3.448 11 4 11 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- CALENDAR ------------------- */

    calendar: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 6.25 9.5 L 7.25 9.5 L 7.25 10.5 L 6.25 10.5 Z M 6.25 13 L 7.25 13 L 7.25 14 L 6.25 14 Z M 6.25 16.5 L 7.25 16.5 L 7.25 17.5 L 6.25 17.5 Z M 9.75 9.5 L 10.75 9.5 L 10.75 10.5 L 9.75 10.5 Z M 9.75 13 L 10.75 13 L 10.75 14 L 9.75 14 Z M 9.75 16.5 L 10.75 16.5 L 10.75 17.5 L 9.75 17.5 Z M 13.25 9.5 L 14.25 9.5 L 14.25 10.5 L 13.25 10.5 Z M 13.25 13 L 14.25 13 L 14.25 14 L 13.25 14 Z M 13.25 16.5 L 14.25 16.5 L 14.25 17.5 L 13.25 17.5 Z M 16.75 9.5 L 17.75 9.5 L 17.75 10.5 L 16.75 10.5 Z M 16.75 13 L 17.75 13 L 17.75 14 L 16.75 14 Z"
                    fill={color}
                    strokeWidth={weight - 1.1}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 22 9 L 22 19 C 22 20.657 20.657 22 19 22 L 5 22 C 3.343 22 2 20.657 2 19 L 2 9"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 22 9 L 22 5 C 22 3.343 20.657 2 19 2 L 5 2 C 3.343 2 2 3.343 2 5 L 2 9 L 2 9 L 2 8 C 2 6.343 3.343 5 5 5 L 19 5 C 20.657 5 22 6.343 22 8 L 22 9 Z"
                    fill={color}
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- SHOPPING BAG ------------------- */

    shoppingBag: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 4 10 C 4 8.343 5.343 7 7 7 L 17 7 C 18.657 7 20 8.343 20 10 L 20 19 C 20 20.657 18.657 22 17 22 L 7 22 C 5.343 22 4 20.657 4 19 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 8 4.5 L 8 4.5 C 8 2.843 9.343 1.5 11 1.5 L 13 1.5 C 14.657 1.5 16 2.843 16 4.5 L 16 4.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- PEN PAPER ------------------- */

    penPaper: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 8 20 L 5 20 C 3.343 20 2 18.657 2 17 L 2 5 C 2 3.343 3.343 2 5 2 L 17 2 C 18.657 2 20 3.343 20 5 L 20 8"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 18.75 12.75 C 19.716 11.784 21.284 11.784 22.25 12.75 L 22.25 12.75 C 23.216 13.716 23.216 15.284 22.25 16.25 L 15.5 23 L 12 23 L 12 19.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 7 7 L 15 7 C 15.276 7 15.5 7.224 15.5 7.5 C 15.5 7.776 15.276 8 15 8 L 7 8 C 6.724 8 6.5 7.776 6.5 7.5 C 6.5 7.224 6.724 7 7 7 Z M 7 11 L 13 11 C 13.276 11 13.5 11.224 13.5 11.5 C 13.5 11.776 13.276 12 13 12 L 7 12 C 6.724 12 6.5 11.776 6.5 11.5 C 6.5 11.224 6.724 11 7 11 Z"
                    fill="hsl(0, 0%, 0%)"
                    strokeWidth={weight - 1.1}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- PEN LINE ------------------- */

    penLine: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 15.293 1.707 C 15.683 1.317 16.317 1.317 16.707 1.707 L 19.293 4.293 C 19.683 4.683 19.683 5.317 19.293 5.707 L 10 15 L 5 16 L 6 11 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 4 21 L 20 21"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- PHOTO ------------------- */

    photo: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 7.204 8.705 C 7.543 7.979 8.556 7.926 8.968 8.613 L 13.13 15.551 C 13.522 16.203 14.469 16.197 14.852 15.54 L 16.569 12.597 C 16.969 11.91 17.971 11.943 18.327 12.653 L 22.276 20.553 C 22.609 21.218 22.125 22 21.382 22 L 2.57 22 C 1.839 22 1.355 21.24 1.664 20.577 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 15 5 C 15 4.448 15.448 4 16 4 L 16 4 C 16.552 4 17 4.448 17 5 L 17 5 C 17 5.552 16.552 6 16 6 L 16 6 C 15.448 6 15 5.552 15 5 Z"
                    fill={color}
                    strokeWidth={weight - 0.5}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- PHOTO - SQUARE ------------------- */

    photoSquare: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 2 5 C 2 3.343 3.343 2 5 2 L 19 2 C 20.657 2 22 3.343 22 5 L 22 19 C 22 20.657 20.657 22 19 22 L 5 22 C 3.343 22 2 20.657 2 19 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 2 15 L 6.483 10.924 C 7.059 10.401 7.938 10.405 8.51 10.932 L 13.601 15.631 C 13.827 15.841 14.176 15.844 14.406 15.639 L 17.521 12.87 C 18.082 12.372 18.925 12.364 19.494 12.852 L 22 15"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 16.5 6.5 C 17.052 6.5 17.5 6.948 17.5 7.5 C 17.5 8.052 17.052 8.5 16.5 8.5 C 15.948 8.5 15.5 8.052 15.5 7.5 C 15.5 6.948 15.948 6.5 16.5 6.5 Z"
                    fill={color}
                    strokeWidth={weight - 0.8}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- SOCIAL ------------------- */

    social: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 8.5 5 L 15.5 5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 19 8.5 L 19 15.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 5 8.5 L 5 15.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 8.5 19 L 15.5 19"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 19 15.5 C 20.933 15.5 22.5 17.067 22.5 19 C 22.5 20.933 20.933 22.5 19 22.5 C 17.067 22.5 15.5 20.933 15.5 19 C 15.5 17.067 17.067 15.5 19 15.5 Z M 5 15.5 C 6.933 15.5 8.5 17.067 8.5 19 C 8.5 20.933 6.933 22.5 5 22.5 C 3.067 22.5 1.5 20.933 1.5 19 C 1.5 17.067 3.067 15.5 5 15.5 Z M 5 1.5 C 6.933 1.5 8.5 3.067 8.5 5 C 8.5 6.933 6.933 8.5 5 8.5 C 3.067 8.5 1.5 6.933 1.5 5 C 1.5 3.067 3.067 1.5 5 1.5 Z M 19 1.5 C 20.933 1.5 22.5 3.067 22.5 5 C 22.5 6.933 20.933 8.5 19 8.5 C 17.067 8.5 15.5 6.933 15.5 5 C 15.5 3.067 17.067 1.5 19 1.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- STAR ------------------- */

    star: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 11.731 1.547 C 11.841 1.324 12.159 1.324 12.269 1.547 L 15.33 7.771 C 15.373 7.86 15.458 7.921 15.555 7.936 L 22.4 8.934 C 22.646 8.97 22.744 9.271 22.567 9.445 L 17.612 14.292 C 17.542 14.361 17.51 14.46 17.526 14.557 L 18.696 21.4 C 18.738 21.645 18.48 21.832 18.26 21.716 L 12.14 18.487 C 12.052 18.44 11.948 18.44 11.86 18.487 L 5.74 21.716 C 5.52 21.832 5.262 21.645 5.304 21.4 L 6.474 14.557 C 6.49 14.46 6.458 14.361 6.388 14.292 L 1.433 9.445 C 1.256 9.271 1.354 8.97 1.6 8.934 L 8.445 7.936 C 8.542 7.921 8.627 7.86 8.67 7.771 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- CONVERSATIONS ------------------- */

    conversations: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 2 4 L 16 4"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 2 10 L 22 10"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 2 16 L 11 16 L 17 21 L 17 16 L 22 16"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- NUMPAD ------------------- */

    numpad: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 11.5 4.5 L 12.5 4.5 L 12.5 5.5 L 11.5 5.5 Z M 11.5 11.5 L 12.5 11.5 L 12.5 12.5 L 11.5 12.5 Z M 11.5 18.5 L 12.5 18.5 L 12.5 19.5 L 11.5 19.5 Z M 4.5 4.5 L 5.5 4.5 L 5.5 5.5 L 4.5 5.5 Z M 4.5 11.5 L 5.5 11.5 L 5.5 12.5 L 4.5 12.5 Z M 18.5 4.5 L 19.5 4.5 L 19.5 5.5 L 18.5 5.5 Z M 18.5 11.5 L 19.5 11.5 L 19.5 12.5 L 18.5 12.5 Z"
                    fill={color}
                    strokeWidth={weight + 0.25}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- CONTACTS ------------------- */

    contacts: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 4 9 L 0 9"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 4 15 L 0 15"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 2 2 L 19 2 C 20.657 2 22 3.343 22 5 L 22 19 C 22 20.657 20.657 22 19 22 L 2 22"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 8 16 C 8 14.5 8.5 13 12 13 C 15.5 13 16 14.5 16 16 C 13 16 11.781 16 8 16 Z M 12 7 C 13.031 7 13.5 7.66 13.5 8.474 C 13.5 9.288 13.312 10.5 12 10.5 C 10.687 10.5 10.5 9.288 10.5 8.474 C 10.5 7.66 10.969 7 12 7 Z"
                    fill={color}
                    strokeWidth={weight - 1.4}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- MENTION ------------------- */

    mention: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 16.415 21.007 C 11.454 23.386 5.497 21.345 3.05 16.428 C 0.603 11.511 2.577 5.546 7.477 3.045 C 10.593 1.503 14.288 1.677 17.244 3.504 C 20.201 5.332 22 8.555 22 12.023"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 16.414 12.023 L 16.414 13.695 C 16.414 15.233 17.664 16.48 19.207 16.48 C 20.75 16.48 22 15.233 22 13.695 L 22 12.023"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 11.945 7.567 C 14.413 7.567 16.414 9.562 16.414 12.023 C 16.414 14.485 14.413 16.48 11.945 16.48 C 9.477 16.48 7.476 14.485 7.476 12.023 C 7.476 9.562 9.477 7.567 11.945 7.567 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- SHARE - iOS ------------------- */

    shareIOS: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 3 13 L 3 19 C 3 20.657 4.343 22 6 22 L 18 22 C 19.657 22 21 20.657 21 19 L 21 13"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 12 15 L 12 2.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 6 8 L 12 2 L 18 8"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- SHARE - Android ------------------- */

    shareAndroid: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 8.5 9 L 15.5 5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 8.5 15 L 15.5 19"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 15 17.5 C 15 16.119 16.119 15 17.5 15 L 19.5 15 C 20.881 15 22 16.119 22 17.5 L 22 19.5 C 22 20.881 20.881 22 19.5 22 L 17.5 22 C 16.119 22 15 20.881 15 19.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 2 11 C 2 9.619 3.119 8.5 4.5 8.5 L 6.5 8.5 C 7.881 8.5 9 9.619 9 11 L 9 13 C 9 14.381 7.881 15.5 6.5 15.5 L 4.5 15.5 C 3.119 15.5 2 14.381 2 13 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 15 4.5 C 15 3.119 16.119 2 17.5 2 L 19.5 2 C 20.881 2 22 3.119 22 4.5 L 22 6.5 C 22 7.881 20.881 9 19.5 9 L 17.5 9 C 16.119 9 15 7.881 15 6.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- SPEAKER ------------------- */

    speaker: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 14 22 L 6 16 L 1 16 L 1 8 L 6 8 L 14 2 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 17.5 15.5 C 19.5 13.5 19.5 10.4 17.5 8.4"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 20.4 18.4 C 23.9 14.9 23.9 9.2 20.4 5.7"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- SLASH - CIRCLE ------------------- */

    slashCircle: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 12 1.5 C 17.799 1.5 22.5 6.201 22.5 12 C 22.5 17.799 17.799 22.5 12 22.5 C 6.201 22.5 1.5 17.799 1.5 12 C 1.5 6.201 6.201 1.5 12 1.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 4.5 19.5 L 19.5 4.5"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- MESSAGE ------------------- */

    message: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 3 5 C 3 3.343 4.343 2 6 2 L 18 2 C 19.657 2 21 3.343 21 5 L 21 15 C 21 16.657 19.657 18 18 18 L 8.592 18 C 8.211 18 7.843 18.141 7.56 18.396 L 3 22.5 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- MESSAGE - FILL ------------------- */

    messageFill: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 2 5 C 2 2.791 3.791 1 6 1 L 18 1 C 20.209 1 22 2.791 22 5 L 22 15 C 22 17.209 20.209 19 18 19 L 8.273 19 C 7.776 19 7.296 19.185 6.928 19.52 L 2.836 23.24 C 2.515 23.532 2 23.304 2 22.87 Z"
                    fill={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- PHONE ------------------- */

    phone: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 6.882 2.623 L 8.68 7.029 C 9.097 8.051 8.712 9.224 7.771 9.8 C 7.477 9.98 7.282 10.285 7.241 10.627 C 7.201 10.97 7.319 11.312 7.563 11.556 L 12.449 16.44 C 12.964 16.956 13.825 16.853 14.205 16.233 C 14.781 15.292 15.955 14.907 16.976 15.325 L 21.383 17.122 C 21.643 17.228 21.845 17.439 21.941 17.702 C 22.036 17.965 22.016 18.257 21.885 18.504 L 21.203 19.792 C 20.284 21.526 18.275 22.38 16.388 21.838 C 9.51 19.866 4.133 14.49 2.161 7.613 C 1.621 5.728 2.475 3.723 4.208 2.805 L 5.499 2.123 C 5.792 1.968 6.038 1.971 6.302 2.067 C 6.565 2.162 6.882 2.623 6.882 2.623 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeMiterlimit="10"
                ></path>
            </svg>
        )
    },

    /* ------------------- PHONE FILL ------------------- */

    phoneFill: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 6.37 1.686 L 8.348 6.532 C 8.806 7.656 8.383 8.946 7.348 9.58 C 7.025 9.778 6.81 10.114 6.766 10.49 C 6.721 10.867 6.851 11.243 7.119 11.511 L 12.493 16.884 C 13.06 17.451 14.007 17.339 14.425 16.656 C 15.059 15.621 16.35 15.198 17.474 15.657 L 22.322 17.635 C 22.607 17.751 22.83 17.983 22.935 18.272 C 23.04 18.562 23.017 18.882 22.873 19.155 L 22.124 20.571 C 21.112 22.479 18.902 23.418 16.827 22.822 C 9.261 20.652 3.347 14.739 1.177 7.175 C 0.583 5.101 1.522 2.895 3.429 1.886 L 4.849 1.135 C 5.171 0.965 5.442 0.968 5.732 1.073 C 6.022 1.179 6.37 1.686 6.37 1.686 Z"
                    fill={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- VOICEMAIL ------------------- */

    voicemail: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 19 8 C 21.209 8 23 9.791 23 12 C 23 14.209 21.209 16 19 16 C 16.791 16 15 14.209 15 12 C 15 9.791 16.791 8 19 8 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 5 8 C 7.209 8 9 9.791 9 12 C 9 14.209 7.209 16 5 16 C 2.791 16 1 14.209 1 12 C 1 9.791 2.791 8 5 8 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 5.5 16 L 12 16 L 18.5 16"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- COG ------------------- */

    cog: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 14.061 1.195 C 12.699 0.935 11.301 0.935 9.939 1.195 L 8.675 4.09 C 8.012 4.369 7.386 4.731 6.813 5.166 L 3.673 4.812 C 2.767 5.862 2.068 7.073 1.612 8.382 L 3.488 10.925 C 3.397 11.639 3.397 12.361 3.488 13.075 L 1.612 15.618 C 2.068 16.927 2.767 18.138 3.673 19.188 L 6.813 18.834 C 7.386 19.269 8.012 19.631 8.675 19.91 L 9.939 22.805 C 11.301 23.065 12.699 23.065 14.061 22.805 L 15.325 19.91 C 15.988 19.631 16.614 19.269 17.187 18.834 L 20.327 19.188 C 21.233 18.138 21.932 16.927 22.388 15.618 L 20.512 13.075 C 20.603 12.361 20.603 11.639 20.512 10.925 L 22.388 8.382 C 21.932 7.073 21.233 5.862 20.327 4.812 L 17.187 5.166 C 16.614 4.731 15.988 4.369 15.325 4.09 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeMiterlimit="10"
                    strokeDasharray=""
                ></path>
                <path
                    d="M 8 12 C 8 9.791 9.791 8 12 8 C 14.209 8 16 9.791 16 12 C 16 14.209 14.209 16 12 16 C 9.791 16 8 14.209 8 12 Z"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeMiterlimit="10"
                    strokeDasharray=""
                ></path>
            </svg>
        )
    },

    /* ------------------- TRASH CAN ------------------- */

    trashCan: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 9 2 L 15 2"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 5 6 L 5.883 20.125 C 5.949 21.179 6.823 22 7.879 22 L 16.121 22 C 17.177 22 18.051 21.179 18.117 20.125 L 19 6"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinecap="round"
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 2.5 6 L 21.5 6"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                ></path>
                <path
                    d="M 10 9 L 10 18"
                    fill="transparent"
                    strokeWidth={weight - 0.5}
                    stroke={color}
                ></path>
                <path
                    d="M 14 9 L 14 18"
                    fill="transparent"
                    strokeWidth={weight - 0.5}
                    stroke={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- COUPON ------------------- */

    coupon: (color: string, weight: number) => {
        return (
            <svg viewBox="0, 0, 24, 24">
                <path
                    d="M 20 7 L 21.5 7 C 22.328 7 23 7.672 23 8.5 L 23 10"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 23 12 L 23 16 L 23 16"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 23 18 L 23 19.5 C 23 20.328 22.328 21 21.5 21 L 20 21"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 18 21 L 14 21"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 12 21 L 8 21"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 18 7 L 14 7"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 6 21 L 4.5 21 C 3.672 21 3 20.328 3 19.5 L 3 18"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
                <path
                    d="M 12.009 2.014 C 12.83 1.455 13.949 1.674 14.5 2.5 L 7.9 7 L 14.5 11.5 C 13.949 12.326 12.83 12.545 12.009 11.986 L 6.29 8.097 L 4.108 9.586 C 4.653 10.044 5 10.732 5 11.5 C 5 12.881 3.881 14 2.5 14 C 1.119 14 0 12.881 0 11.5 C 0 10.596 0.479 9.805 1.198 9.366 L 4.676 7 L 1.198 4.634 C 0.479 4.195 0 3.404 0 2.5 C 0 1.119 1.119 -0 2.5 -0 C 3.881 -0 5 1.119 5 2.5 C 5 3.268 4.653 3.956 4.108 4.414 L 6.29 5.903 Z M 2.5 12.5 C 3.052 12.5 3.5 12.052 3.5 11.5 C 3.5 10.948 3.052 10.5 2.5 10.5 C 1.948 10.5 1.5 10.948 1.5 11.5 C 1.5 12.052 1.948 12.5 2.5 12.5 Z M 1.5 2.5 C 1.5 3.052 1.948 3.5 2.5 3.5 C 3.052 3.5 3.5 3.052 3.5 2.5 C 3.5 1.948 3.052 1.5 2.5 1.5 C 1.948 1.5 1.5 1.948 1.5 2.5 Z"
                    fill={color}
                ></path>
            </svg>
        )
    },

    /* ------------------- CARET DOWN ------------------- */

    caretDown12: (color: string, weight: number) => {
        return (
            <svg width="12" height="12">
                <path
                    d="M 2 4 L 6 8 L 10 4"
                    fill="transparent"
                    strokeWidth={weight}
                    stroke={color}
                    strokeLinejoin="round"
                ></path>
            </svg>
        )
    },

    /* ------------------- CLEAR (20) ------------------- */

    clear20: (color: string, weight: number) => {
        return (
            <svg width="20" height="20">
                <path
                    d="M 10 0 C 15.523 0 20 4.477 20 10 C 20 15.523 15.523 20 10 20 C 4.477 20 0 15.523 0 10 C 0 4.477 4.477 0 10 0 Z M 10 8.586 L 7.172 5.757 L 5.757 7.172 L 8.586 10 L 5.757 12.828 L 7.172 14.243 L 10 11.414 L 12.828 14.243 L 14.243 12.828 L 11.414 10 L 14.243 7.172 L 12.828 5.757 Z"
                    fill={color}
                ></path>
            </svg>
        )
    },
}
