// @framer/godaddy.maui/code/Foundation/Overlay.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"
import { themeColors } from "./Shared"

/* ----------------------------------------------------------------------------------------- */
/*                                  OVERLAY COMPONENT                                        */
/* ----------------------------------------------------------------------------------------- */

export function Overlay({ ...props }) {
    return (
        <Frame
            size="100%"
            background={
                props.mode === "light"
                    ? themeColors.light.bgOverlay
                    : themeColors.dark.bgOverlay
            }
            {...props}
        />
    )
}

Overlay.defaultProps = {
    width: 375,
    height: 812,
    mode: "light",
}

// Learn more: https://framer.com/api/property-controls/
addPropertyControls(Overlay, {
    mode: {
        title: "Mode",
        type: ControlType.SegmentedEnum,
        options: ["light", "dark"],
        optionTitles: ["Light", "Dark"],
    },
})
