// @framer/godaddy.maui/code/Platform/Home_Indicator.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"

export function Home_Indicator({ ...props }) {
    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                   CONTENT                                                     */
    /* ------------------------------------------------------------------------------------------------------------- */
    const renderHomeIndicatorContent = () => {
        switch (props.type) {
            case "iOSX":
                return (
                    <Frame
                        width={135}
                        height={5}
                        bottom={8}
                        background={props.mode === "light" ? "#000" : "#FFF"}
                        radius={100}
                        center="x"
                    />
                )
                break
            case "pixel3":
                return (
                    <Frame
                        width={75}
                        height={3}
                        bottom={7}
                        background={props.mode === "light" ? "#000" : "#FFF"}
                        radius={100}
                        center="x"
                    />
                )
                break
        }
    }

    const homeIndicatorContent = React.useMemo(
        () => renderHomeIndicatorContent(),
        [props]
    )

    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                   HEIGHT                                                      */
    /* ------------------------------------------------------------------------------------------------------------- */

    const getHomeIndicatorHeight = () => {
        switch (props.type) {
            case "iOSX":
                return 34
                break
            case "pixel3":
                return 17
                break
        }
    }

    const homeIndicatorHeight = React.useMemo(() => getHomeIndicatorHeight(), [
        props,
    ])

    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                    MAIN                                                       */
    /* ------------------------------------------------------------------------------------------------------------- */

    return (
        <Frame
            {...props}
            bottom={0}
            width="100%"
            height={homeIndicatorHeight}
            style={{
                zIndex: 1000,
            }}
        >
            {homeIndicatorContent}
        </Frame>
    )
}

/* ------------------------------------------------------------------------------------------------------------- */
/*                                                   PROPERTIES                                                  */
/* ------------------------------------------------------------------------------------------------------------- */

Home_Indicator.defaultProps = {
    width: 375,
    height: 34,
    type: "iOSX",
    background: "none",
    visible: true,
    opacity: 1,
    transition: {},
}

addPropertyControls(Home_Indicator, {
    type: {
        title: "Type",
        type: ControlType.Enum,
        options: ["iOSX", "pixel3"],
        optionTitles: ["iOS X (34)", "Pixel 3 (17)"],
    },
    background: {
        title: "Bg",
        type: ControlType.Color,
    },
    mode: {
        title: "Mode",
        type: ControlType.SegmentedEnum,
        options: ["light", "dark"],
        optionTitles: ["Light", "Dark"],
    },
})
