// @framer/godaddy.maui/code/Platform/Keyboard.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType, Stack } from "framer"

// SF Font
const SFProText =
    ".SFNSText, SFProText-Regular, SFUIText-Regular, .SFUIText, -apple-system, BlinkMacSystemFont, sans-serif"

function Key_Shift(props) {
    return (
        <Frame
            width={42}
            height={42}
            radius={5}
            background="##ADB3BC"
            shadow="0px 1px 0px 0px #898A8D"
        >
            <Frame size={28} background="none" center>
                <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28">
                    <path
                        d="M 22.254 15.492 L 18.023 15.492 L 18.023 19.472 C 18.023 19.552 18.023 19.635 18.023 19.702 C 18.023 19.769 18.023 19.833 18.02 19.899 C 18.018 20.043 18.006 20.186 17.983 20.327 C 17.958 20.468 17.914 20.604 17.848 20.731 C 17.783 20.859 17.697 20.975 17.594 21.076 C 17.491 21.176 17.374 21.261 17.245 21.326 C 17.115 21.389 16.977 21.434 16.835 21.46 C 16.692 21.482 16.548 21.494 16.404 21.497 C 16.337 21.497 16.272 21.5 16.206 21.5 L 10.762 21.5 C 10.695 21.5 10.63 21.497 10.564 21.497 C 10.42 21.494 10.276 21.482 10.133 21.46 C 9.991 21.434 9.853 21.389 9.723 21.326 C 9.595 21.261 9.477 21.176 9.374 21.076 C 9.272 20.975 9.184 20.859 9.12 20.731 C 9.056 20.604 9.011 20.468 8.986 20.327 C 8.963 20.186 8.95 20.043 8.948 19.899 C 8.948 19.833 8.945 19.779 8.945 19.702 C 8.945 19.625 8.945 19.552 8.945 19.472 L 8.945 15.492 L 4.647 15.492 C 4.213 15.492 3.733 14.48 4.177 14.042 C 4.621 13.604 11.596 6.687 13.056 5.247 C 13.163 5.116 13.315 5.028 13.483 5 L 13.494 5 C 13.639 5.008 13.774 5.077 13.864 5.19 C 15.323 6.631 22.245 13.524 22.767 14.042 C 23.289 14.56 22.852 15.492 22.254 15.492 Z"
                        fill="transparent"
                        strokeWidth="1.3"
                        stroke="rgba(0,0,0,1)"
                        strokeMiterlimit="10"
                    ></path>
                </svg>
            </Frame>
        </Frame>
    )
}

//  ------------------------------------------------------------------
//  DELETE

function Key_Delete(props) {
    return (
        <Frame
            width={42}
            height={42}
            radius={5}
            background="#ADB3BC"
            shadow="0px 1px 0px 0px #898A8D"
            onTap={props.onTap}
            overflow="hidden"
        >
            <Frame size={28} background="none" center>
                <svg xmlns="http://www.w3.org/2000/svg" width="28" height="28">
                    <path
                        d="M 19.706 11.605 L 17.161 14.153 L 19.644 16.633 C 19.667 16.649 19.688 16.667 19.709 16.687 C 19.871 16.849 19.919 17.091 19.832 17.303 C 19.745 17.514 19.539 17.652 19.311 17.652 C 19.158 17.652 19.016 17.59 18.912 17.487 L 18.906 17.493 L 16.362 14.952 L 13.821 17.496 L 13.813 17.488 C 13.762 17.538 13.701 17.579 13.633 17.607 C 13.421 17.695 13.177 17.646 13.015 17.484 C 12.795 17.263 12.795 16.905 13.015 16.684 C 13.047 16.652 13.082 16.624 13.12 16.601 L 15.564 14.154 L 13.015 11.608 L 13.023 11.6 C 12.973 11.549 12.932 11.488 12.904 11.42 C 12.816 11.208 12.865 10.964 13.028 10.802 C 13.249 10.582 13.606 10.582 13.827 10.802 C 13.859 10.834 13.887 10.87 13.91 10.907 L 16.362 13.355 L 18.837 10.878 C 18.856 10.851 18.877 10.826 18.9 10.802 C 19.061 10.64 19.304 10.592 19.515 10.679 C 19.726 10.766 19.864 10.972 19.864 11.2 C 19.864 11.353 19.803 11.495 19.7 11.599 Z M 6.271 9.726 L 8.406 7.588 L 8.727 7.267 C 8.804 7.19 8.866 7.129 9.006 6.995 C 9.006 6.995 9.043 6.957 9.051 6.949 C 9.086 6.912 9.107 6.89 9.12 6.877 C 9.411 6.586 9.745 6.341 10.119 6.148 C 10.485 5.966 10.874 5.838 11.285 5.765 C 11.657 5.706 12.033 5.674 12.392 5.67 C 12.534 5.664 12.645 5.663 12.879 5.663 C 13.04 5.661 13.169 5.66 13.452 5.66 L 19.775 5.66 C 19.917 5.66 19.98 5.66 20.059 5.66 C 20.138 5.661 20.258 5.662 20.338 5.663 C 20.583 5.663 20.693 5.664 20.818 5.669 C 21.195 5.674 21.572 5.706 21.955 5.767 C 22.355 5.838 22.743 5.966 23.11 6.15 C 23.477 6.334 23.816 6.584 24.106 6.877 C 24.397 7.169 24.641 7.502 24.832 7.875 C 25.023 8.248 25.142 8.63 25.215 9.042 C 25.274 9.413 25.306 9.789 25.31 10.148 C 25.313 10.224 25.315 10.3 25.316 10.401 C 25.318 10.503 25.317 10.442 25.317 10.482 C 25.318 10.521 25.319 10.594 25.32 10.648 L 25.32 17.638 C 25.319 17.68 25.319 17.715 25.317 17.794 C 25.316 17.873 25.314 18.035 25.311 18.113 C 25.306 18.495 25.273 18.875 25.21 19.254 C 25.148 19.633 25.013 20.053 24.826 20.427 C 24.639 20.802 24.391 21.133 24.098 21.427 C 23.806 21.72 23.465 21.969 23.094 22.158 C 22.722 22.347 22.33 22.476 21.92 22.545 C 21.509 22.615 21.166 22.641 20.788 22.648 C 20.718 22.649 20.664 22.65 20.543 22.651 C 20.421 22.652 20.374 22.653 20.297 22.654 L 12.876 22.654 C 12.814 22.653 12.763 22.652 12.647 22.651 C 12.531 22.65 12.473 22.649 12.408 22.648 C 12.031 22.642 11.655 22.609 11.276 22.547 C 10.874 22.476 10.485 22.348 10.11 22.162 C 9.745 21.973 9.411 21.728 9.121 21.438 C 9.107 21.424 9.086 21.402 9.05 21.364 C 9.043 21.357 9.021 21.334 9.006 21.319 C 8.866 21.185 8.804 21.124 8.727 21.047 C 8.727 21.047 6.237 18.559 6.226 18.549 C 6.168 18.494 6.11 18.436 6.054 18.378 C 6.003 18.328 5.958 18.284 5.866 18.193 L 3.912 16.235 L 3.727 16.05 C 3.676 15.999 3.639 15.96 3.576 15.893 C 3.411 15.73 3.254 15.559 3.107 15.379 C 2.971 15.233 2.861 15.065 2.783 14.88 L 2.764 14.829 C 2.623 14.392 2.623 13.922 2.781 13.438 C 2.859 13.252 2.969 13.081 3.107 12.935 C 3.254 12.754 3.411 12.582 3.559 12.439 C 3.639 12.354 3.676 12.315 3.727 12.264 C 3.777 12.214 3.856 12.134 3.912 12.076 L 6.055 9.932 C 6.094 9.894 6.113 9.875 6.136 9.852 C 6.16 9.829 6.198 9.792 6.213 9.778 C 6.233 9.758 6.271 9.726 6.271 9.726 Z M 13.452 6.919 C 13.176 6.919 13.052 6.919 12.889 6.922 C 12.661 6.922 12.562 6.923 12.424 6.928 C 12.109 6.932 11.795 6.959 11.494 7.007 C 11.211 7.057 10.937 7.147 10.689 7.27 C 10.439 7.4 10.21 7.567 10.011 7.767 C 10.009 7.768 10 7.778 9.96 7.82 C 9.913 7.869 9.892 7.891 9.868 7.912 C 9.742 8.034 9.685 8.089 9.617 8.157 L 9.297 8.478 L 7.125 10.653 C 7.102 10.676 7.079 10.696 7.02 10.749 C 6.961 10.802 6.983 10.785 6.946 10.823 L 6.757 11.011 L 4.814 12.954 C 4.759 13.012 4.715 13.056 4.617 13.155 C 4.518 13.253 4.547 13.226 4.456 13.321 C 4.319 13.454 4.191 13.596 4.071 13.745 L 4.028 13.793 C 3.998 13.824 3.972 13.86 3.953 13.899 C 3.905 14.067 3.905 14.245 3.953 14.413 C 3.973 14.454 3.999 14.491 4.031 14.524 L 4.069 14.567 C 4.191 14.717 4.321 14.86 4.476 15.013 C 4.547 15.089 4.578 15.121 4.617 15.16 L 4.803 15.346 L 6.757 17.303 C 6.847 17.393 6.887 17.432 6.959 17.502 C 6.994 17.539 7.031 17.577 7.076 17.62 C 7.085 17.628 7.099 17.641 7.122 17.662 L 9.617 20.157 C 9.685 20.225 9.742 20.281 9.868 20.402 C 9.892 20.423 9.913 20.445 9.962 20.496 C 10 20.537 10.009 20.546 10.011 20.548 C 10.21 20.747 10.439 20.915 10.68 21.04 C 10.937 21.167 11.211 21.258 11.486 21.306 C 11.797 21.357 12.111 21.385 12.43 21.389 C 12.492 21.39 12.543 21.391 12.659 21.392 C 12.775 21.393 12.833 21.394 12.889 21.395 L 20.285 21.395 C 20.355 21.394 20.409 21.393 20.53 21.392 C 20.651 21.391 20.699 21.39 20.764 21.389 C 21.083 21.384 21.401 21.355 21.712 21.303 C 22.024 21.252 22.268 21.165 22.524 21.036 C 22.779 20.906 23.006 20.74 23.206 20.539 C 23.406 20.338 23.576 20.105 23.703 19.858 C 23.83 19.611 23.921 19.326 23.969 19.045 C 24.017 18.765 24.048 18.417 24.052 18.082 C 24.055 18 24.057 17.932 24.059 17.777 C 24.061 17.622 24.06 17.659 24.061 17.625 L 24.061 10.661 C 24.06 10.615 24.06 10.579 24.059 10.498 C 24.058 10.418 24.058 10.459 24.058 10.42 C 24.057 10.381 24.055 10.261 24.052 10.181 C 24.048 9.865 24.021 9.551 23.973 9.25 C 23.923 8.967 23.833 8.693 23.709 8.443 C 23.584 8.192 23.415 7.965 23.215 7.765 C 23.015 7.565 22.789 7.399 22.542 7.274 C 22.296 7.148 22.016 7.056 21.747 7.009 C 21.434 6.959 21.119 6.932 20.785 6.928 C 20.665 6.923 20.566 6.922 20.328 6.922 C 20.243 6.92 20.154 6.92 20.054 6.919 C 19.953 6.919 19.915 6.919 19.775 6.919 Z"
                        fill="rgba(0,0,0,1)"
                    ></path>
                </svg>
            </Frame>
        </Frame>
    )
}

function Key_123(props) {
    return (
        <Frame
            width={87}
            height={42}
            radius={5}
            background="#ADB3BC"
            shadow="0px 1px 0px 0px #898A8D"
        >
            <Frame
                width="100%"
                height={28}
                top={6}
                background="none"
                style={{
                    fontFamily: SFProText,
                    fontSize: "15px",
                    lineHeight: "4",
                    textAlign: "center",
                }}
            >
                123
            </Frame>
        </Frame>
    )
}
function Key_Space(props) {
    return (
        <Frame
            width={87}
            height={42}
            radius={5}
            background="#FCFCFE"
            shadow="0px 1px 0px 0px #898A8D"
        >
            <Frame
                width="100%"
                height={28}
                top={5}
                background="none"
                style={{
                    fontFamily: SFProText,
                    fontSize: "17px",
                    lineHeight: "4",
                    textAlign: "center",
                }}
            >
                space
            </Frame>
        </Frame>
    )
}

//  ------------------------------------------------------------------
//  AT KEY

function Key_At(props) {
    return (
        <Frame
            width={42}
            height={42}
            radius={5}
            background="#FCFCFE"
            shadow="0px 1px 0px 0px #898A8D"
            onTap={props.onTap}
            overflow="hidden"
        >
            <Frame
                width="100%"
                height={28}
                top={5}
                background="none"
                style={{
                    fontFamily: SFProText,
                    fontSize: "17px",
                    lineHeight: "4",
                    textAlign: "center",
                }}
            >
                @
            </Frame>
        </Frame>
    )
}

//  ------------------------------------------------------------------
//  DOT

function Key_Dot(props) {
    return (
        <Frame
            width={42}
            height={42}
            radius={5}
            background="#FCFCFE"
            shadow="0px 1px 0px 0px #898A8D"
            onTap={props.onTap}
            overflow="hidden"
        >
            <Frame
                width="100%"
                height={28}
                top={5}
                background="none"
                style={{
                    fontFamily: SFProText,
                    fontSize: "17px",
                    lineHeight: "4",
                    textAlign: "center",
                }}
            >
                .
            </Frame>
        </Frame>
    )
}

//  ------------------------------------------------------------------
//  RETURN

function Key_Return(props) {
    return (
        <Frame
            width={87}
            height={42}
            radius={5}
            background="#FCFCFE"
            shadow="0px 1px 0px 0px #898A8D"
        >
            <Frame
                width="100%"
                height={28}
                top={5}
                background="none"
                style={{
                    fontFamily: SFProText,
                    fontSize: "17px",
                    lineHeight: "4",
                    textAlign: "center",
                }}
            >
                {props.content}
            </Frame>
        </Frame>
    )
}

function Key_Emoji(props) {
    return (
        <Frame width={88} height={51} background="#none">
            <Frame size={29} center background="none">
                <svg xmlns="http://www.w3.org/2000/svg" width="29" height="29">
                    <path
                        d="M 10.229 8.644 C 9.812 8.639 9.41 8.802 9.114 9.096 C 8.817 9.389 8.651 9.79 8.652 10.207 C 8.652 11.078 9.358 11.784 10.229 11.784 C 11.1 11.784 11.806 11.078 11.806 10.207 C 11.808 9.79 11.641 9.389 11.345 9.096 C 11.048 8.802 10.647 8.639 10.229 8.644 Z M 18.78 8.644 C 18.362 8.639 17.96 8.802 17.664 9.096 C 17.367 9.389 17.201 9.79 17.202 10.207 C 17.194 10.776 17.493 11.305 17.984 11.592 C 18.475 11.879 19.083 11.879 19.575 11.592 C 20.066 11.305 20.365 10.776 20.357 10.207 C 20.358 9.79 20.191 9.389 19.895 9.096 C 19.599 8.802 19.197 8.639 18.78 8.644 Z M 14.5 1 C 21.956 1 28 7.044 28 14.5 C 28 21.956 21.956 28 14.5 28 C 7.044 28 1 21.956 1 14.5 C 1 7.044 7.044 1 14.5 1 Z M 14.5 2.573 C 11.336 2.572 8.302 3.829 6.065 6.066 C 3.828 8.303 2.571 11.336 2.571 14.5 C 2.571 21.087 7.911 26.427 14.498 26.427 C 21.085 26.427 26.425 21.088 26.426 14.501 C 26.426 7.914 21.087 2.573 14.5 2.573 Z M 14.498 23.953 C 10.004 23.953 6.277 20.757 5.516 16.916 C 5.362 15.958 6.147 15.732 6.886 15.911 C 9.36 16.626 11.925 16.974 14.5 16.943 C 17.069 16.973 19.629 16.626 22.097 15.914 C 22.816 15.732 23.586 15.938 23.497 16.825 C 22.778 20.71 18.992 23.953 14.498 23.953 Z M 21.066 17.365 C 21.064 17.366 21.062 17.367 21.06 17.367 C 21.065 17.366 21.067 17.365 21.066 17.365 Z M 14.511 18.212 C 11.869 18.212 9.506 17.919 7.88 17.335 C 6.701 16.984 6.928 18.371 7.756 18.796 C 9.876 19.761 12.182 20.248 14.511 20.221 C 17.281 20.221 19.749 19.576 21.38 18.742 C 22.151 18.246 22.304 16.916 21.056 17.365 C 21.047 17.369 21.038 17.372 21.029 17.376 C 18.906 17.956 16.712 18.238 14.511 18.212 Z"
                        fill="rgba(80,85,92,1)"
                    ></path>
                </svg>
            </Frame>
        </Frame>
    )
}

//  ------------------------------------------------------------------
//  LETTERS

function Key_Letters(props) {
    return (
        <Frame
            width={32}
            height={42}
            radius={5}
            background="#FCFCFE"
            shadow="0px 1px 0px 0px #898A8D"
            overflow="hidden"
        >
            <Frame
                width="100%"
                height={28}
                top={5}
                background="none"
                style={{
                    fontFamily: SFProText,
                    fontSize: "24px",
                    lineHeight: "4",
                    textAlign: "center",
                }}
                onTap={props.onTap}
            >
                {props.content}
            </Frame>
        </Frame>
    )
}

const letters_rowOne = ["q", "w", "e", "r", "t", "y", "u", "i", "o", "p"]
const letters_rowTwo = ["a", "s", "d", "f", "g", "h", "j", "k", "l"]
const letters_rowThree = ["z", "x", "c", "v", "b", "n", "m"]

//  ------------------------------------------------------------------
//  KEYBOARD COMPONENT

export function Keyboard(props) {
    return (
        <Frame size="100%" background="#D1D5DB">
            {/* Row 1 */}
            <Stack
                top={8}
                left={3}
                right={3}
                height={42}
                direction="horizontal"
                distribution="space-between"
            >
                {letters_rowOne.map((letter, index) => {
                    return (
                        <Key_Letters
                            content={letter}
                            key={index.toString()}
                            onTap={() => {
                                props.onKeyTap(letter)
                            }}
                        />
                    )
                })}
            </Stack>

            {/* Row 2 */}
            <Stack
                top={62}
                left={22}
                right={22}
                height={42}
                direction="horizontal"
                distribution="space-between"
            >
                {letters_rowTwo.map((letter, index) => {
                    return (
                        <Key_Letters
                            content={letter}
                            key={index.toString()}
                            onTap={() => {
                                props.onKeyTap(letter)
                            }}
                        />
                    )
                })}
            </Stack>

            {/* Row 3 */}
            <Stack
                top={116}
                left={3}
                right={3}
                height={42}
                direction="horizontal"
                distribution="space-between"
            >
                <Key_Shift />
                <Stack
                    width={257}
                    height={42}
                    direction="horizontal"
                    distribution="space-between"
                >
                    {letters_rowThree.map((letter, index) => {
                        return (
                            <Key_Letters
                                content={letter}
                                key={index.toString()}
                                onTap={() => {
                                    props.onKeyTap(letter)
                                }}
                            />
                        )
                    })}
                </Stack>
                <Key_Delete
                    onTap={() => {
                        props.onKeyTap("delete")
                    }}
                />
            </Stack>

            {/* Row 4 */}
            <Stack
                top={170}
                left={3}
                right={3}
                height={42}
                direction="horizontal"
                distribution="space-between"
            >
                <Key_123 />
                <Key_Space />
                <Key_At
                    onTap={() => {
                        props.onKeyTap("@")
                    }}
                />
                <Key_Dot
                    onTap={() => {
                        props.onKeyTap(".")
                    }}
                />
                <Key_Return content="go" />
            </Stack>

            {/* Row 5 */}
            <Stack
                top={224}
                left={3}
                right={3}
                height={51}
                direction="horizontal"
                distribution="space-between"
            >
                <Key_Emoji />
            </Stack>
        </Frame>
    )
}

//  ------------------------------------------------------------------
//  PROPS

Keyboard.defaultProps = {
    width: 375,
    height: 291,
    onKeyTap: () => {},
}
