// @framer/godaddy.maui/code/Platform/Status_Bar.tsx
import * as React from "react"
import { Frame, addPropertyControls, ControlType } from "framer"

export function Status_Bar(props) {
    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                   CONTENT                                                     */
    /* ------------------------------------------------------------------------------------------------------------- */

    const renderStatusBarContent = () => {
        switch (props.type) {
            //
            //  ----------------------------------------  iOS X  ----------------------------------------
            //
            case "iOSX":
                return (
                    <Frame width="100%" height="100%" background="none">
                        {/* Time */}
                        <Frame
                            width={34}
                            height={12}
                            top={17}
                            left={30}
                            background="none"
                        >
                            <svg width="34px" height="12px" viewBox="0 0 34 12">
                                <g
                                    stroke="none"
                                    strokeWidth="1"
                                    fill="none"
                                    fillRule="evenodd"
                                >
                                    <g
                                        transform="translate(-31.000000, -17.000000)"
                                        fill={
                                            props.mode === "light"
                                                ? "#000"
                                                : "#FFF"
                                        }
                                        fillRule="nonzero"
                                    >
                                        <path d="M35.7973633,28.2563477 C33.8637695,28.2563477 32.3842773,27.1210938 32.0766602,25.3925781 L33.9589844,25.3925781 C34.1860352,26.1835938 34.9111328,26.6743164 35.8193359,26.6743164 C37.3647461,26.6743164 38.2509766,25.2607422 38.2729492,22.8144531 L38.1484375,22.8144531 C37.6503906,23.8178711 36.6469727,24.3891602 35.4091797,24.3891602 C33.3950195,24.3891602 31.9301758,22.909668 31.9301758,20.8808594 C31.9301758,18.7128906 33.5561523,17.1674805 35.855957,17.1674805 C38.3828125,17.1674805 40.0820312,19.1010742 40.0820312,22.5947266 C40.0820312,26.1542969 38.4853516,28.2563477 35.7973633,28.2563477 Z M35.8632812,22.8950195 C37.0864258,22.8950195 37.9580078,22.0380859 37.9580078,20.8588867 C37.9580078,19.6577148 37.0644531,18.7421875 35.8706055,18.7421875 C34.6767578,18.7421875 33.7832031,19.6430664 33.7832031,20.8222656 C33.7832031,22.0454102 34.6328125,22.8950195 35.8632812,22.8950195 Z M43.3999023,26.5424805 C42.7114258,26.5424805 42.2280273,26.0444336 42.2280273,25.3852539 C42.2280273,24.71875 42.7114258,24.2280273 43.3999023,24.2280273 C44.0957031,24.2280273 44.5717773,24.71875 44.5717773,25.3852539 C44.5717773,26.0444336 44.0957031,26.5424805 43.3999023,26.5424805 Z M43.3999023,21.1958008 C42.7114258,21.1958008 42.2280273,20.7050781 42.2280273,20.0458984 C42.2280273,19.3793945 42.7114258,18.8886719 43.3999023,18.8886719 C44.0957031,18.8886719 44.5717773,19.3793945 44.5717773,20.0458984 C44.5717773,20.7050781 44.0957031,21.1958008 43.3999023,21.1958008 Z M49.3544922,23.2978516 L49.3544922,21.8549805 L50.6435547,21.8549805 C51.7275391,21.8549805 52.4746094,21.1958008 52.4746094,20.2509766 C52.4746094,19.3867188 51.8081055,18.7055664 50.6875,18.7055664 C49.5375977,18.7055664 48.8198242,19.2915039 48.7392578,20.331543 L46.9667969,20.331543 C47.0620117,18.4638672 48.4975586,17.1674805 50.7900391,17.1674805 C53.0385742,17.1674805 54.3056641,18.4711914 54.2983398,19.9946289 C54.2910156,21.269043 53.4926758,22.1186523 52.3720703,22.4116211 L52.3720703,22.5361328 C53.8076172,22.7558594 54.7084961,23.6860352 54.7084961,25.121582 C54.7084961,26.9453125 53.0092773,28.2636719 50.7241211,28.2636719 C48.4243164,28.2636719 46.8496094,27.0039062 46.7617188,25.0776367 L48.5708008,25.0776367 C48.6513672,26.0517578 49.4790039,26.6743164 50.7021484,26.6743164 C51.9179688,26.6743164 52.7749023,25.9931641 52.7749023,24.9824219 C52.7749023,23.9423828 51.9838867,23.2978516 50.6801758,23.2978516 L49.3544922,23.2978516 Z M60.5678711,28.2636719 C58.0336914,28.2636719 56.5322266,26.2128906 56.5322266,22.7045898 C56.5322266,19.2109375 58.0556641,17.1674805 60.5678711,17.1674805 C63.0727539,17.1674805 64.5961914,19.2109375 64.5961914,22.7192383 C64.5961914,26.2128906 63.0947266,28.2636719 60.5678711,28.2636719 Z M60.5678711,26.6962891 C61.9082031,26.6962891 62.6845703,25.3266602 62.6845703,22.7045898 C62.6845703,20.097168 61.8935547,18.7421875 60.5678711,18.7421875 C59.2421875,18.7421875 58.4438477,20.1044922 58.4438477,22.7045898 C58.4438477,25.3266602 59.2275391,26.6962891 60.5678711,26.6962891 Z"></path>
                                    </g>
                                </g>
                            </svg>
                        </Frame>

                        {/*Cellular*/}
                        <Frame
                            width={17}
                            height={12}
                            top={17}
                            right={64}
                            background="none"
                        >
                            <svg>
                                <path
                                    d="M 1 7.333 L 2 7.333 C 2.552 7.333 3 7.781 3 8.333 L 3 10.333 C 3 10.886 2.552 11.333 2 11.333 L 1 11.333 C 0.448 11.333 0 10.886 0 10.333 L 0 8.333 C 0 7.781 0.448 7.333 1 7.333 Z M 5.667 5.333 L 6.667 5.333 C 7.219 5.333 7.667 5.781 7.667 6.333 L 7.667 10.333 C 7.667 10.886 7.219 11.333 6.667 11.333 L 5.667 11.333 C 5.114 11.333 4.667 10.886 4.667 10.333 L 4.667 6.333 C 4.667 5.781 5.114 5.333 5.667 5.333 Z M 10.333 3 L 11.333 3 C 11.886 3 12.333 3.448 12.333 4 L 12.333 10.333 C 12.333 10.886 11.886 11.333 11.333 11.333 L 10.333 11.333 C 9.781 11.333 9.333 10.886 9.333 10.333 L 9.333 4 C 9.333 3.448 9.781 3 10.333 3 Z M 15 0.667 L 16 0.667 C 16.552 0.667 17 1.114 17 1.667 L 17 10.333 C 17 10.886 16.552 11.333 16 11.333 L 15 11.333 C 14.448 11.333 14 10.886 14 10.333 L 14 1.667 C 14 1.114 14.448 0.667 15 0.667 Z"
                                    fill={
                                        props.mode === "light" ? "#000" : "#FFF"
                                    }
                                />
                            </svg>
                        </Frame>

                        {/* Wifi */}
                        <Frame
                            width={16}
                            height={12}
                            top={17}
                            right={43}
                            background="none"
                        >
                            <svg>
                                <path
                                    d="M 7.997 2.593 C 10.222 2.593 12.361 3.448 13.974 4.979 C 14.095 5.097 14.289 5.096 14.409 4.976 L 15.569 3.805 C 15.63 3.745 15.664 3.662 15.663 3.576 C 15.663 3.49 15.628 3.408 15.567 3.348 C 11.335 -0.704 4.659 -0.704 0.427 3.348 C 0.365 3.408 0.331 3.49 0.33 3.576 C 0.329 3.662 0.363 3.745 0.424 3.805 L 1.585 4.976 C 1.704 5.096 1.898 5.098 2.02 4.979 C 3.632 3.447 5.772 2.593 7.997 2.593 Z M 7.997 6.401 C 9.219 6.401 10.398 6.855 11.304 7.675 C 11.427 7.791 11.62 7.788 11.739 7.669 L 12.899 6.499 C 12.96 6.437 12.993 6.354 12.993 6.267 C 12.992 6.181 12.956 6.098 12.894 6.038 C 10.135 3.474 5.862 3.474 3.103 6.038 C 3.04 6.098 3.005 6.181 3.004 6.267 C 3.003 6.354 3.037 6.437 3.098 6.499 L 4.257 7.669 C 4.377 7.788 4.57 7.791 4.692 7.675 C 5.598 6.855 6.776 6.401 7.997 6.401 Z M 10.225 9.194 C 10.287 9.133 10.321 9.05 10.319 8.963 C 10.318 8.876 10.28 8.794 10.216 8.736 C 8.935 7.653 7.059 7.653 5.779 8.736 C 5.714 8.794 5.677 8.876 5.675 8.963 C 5.673 9.049 5.707 9.133 5.769 9.194 L 7.775 11.216 C 7.833 11.275 7.913 11.309 7.997 11.309 C 8.081 11.309 8.161 11.275 8.22 11.216 Z"
                                    fill={
                                        props.mode === "light" ? "#000" : "#FFF"
                                    }
                                />
                            </svg>
                        </Frame>

                        {/* Battery */}
                        <Frame
                            width={25}
                            height={12}
                            top={17}
                            right={12}
                            background="none"
                        >
                            <svg>
                                <g fill="none" fillRule="evenodd">
                                    <path
                                        d="m3.7518137.83333333c-1.00827318 0-1.40697456.07699279-1.8185085.29708355-.34743368.18580957-.61711209.45548797-.80292165.80292165-.22009076.41153394-.29708355.81023533-.29708355 1.81850851v4.49630592c0 1.00827318.07699279 1.40697457.29708355 1.81850854.18580956.3474336.45548797.6171121.80292165.8029216.41153394.2200908.81023532.2970836 1.8185085.2970836h15.1629726c1.0082732 0 1.4069746-.0769928 1.8185085-.2970836.3474337-.1858095.6171121-.455488.8029217-.8029216.2200907-.41153397.2970835-.81023536.2970835-1.81850854v-4.49630592c0-1.00827318-.0769928-1.40697457-.2970835-1.81850851-.1858096-.34743368-.455488-.61711208-.8029217-.80292165-.4115339-.22009076-.8102353-.29708355-1.8185085-.29708355z"
                                        opacity=".35"
                                        stroke={
                                            props.mode === "light"
                                                ? "#000"
                                                : "#FFF"
                                        }
                                    />
                                    <g
                                        fill={
                                            props.mode === "light"
                                                ? "#000"
                                                : "#FFF"
                                        }
                                        fillRule="nonzero"
                                    >
                                        <path
                                            d="m23.3333 4v4c.8047311-.33877652 1.328038-1.12686672 1.328038-2s-.5233069-1.66122348-1.328038-2"
                                            opacity=".4"
                                        />
                                        <path d="m4.04255685 2.33333333h14.58148625c.5943459 0 .8098701.06188375 1.027154.17808848.217284.11620473.3878097.28673045.5040144.50401441.1162048.21728396.1780885.43280812.1780885 1.02715396v3.91481964c0 .59434584-.0618837.80987-.1780885 1.02715396-.1162047.21728396-.2867304.38780968-.5040144.50401441-.2172839.11620473-.4328081.17808848-1.027154.17808848h-14.58148625c-.59434584 0-.80987001-.06188375-1.02715397-.17808848-.21728395-.11620473-.38780967-.28673045-.50401441-.50401441-.11620473-.21728396-.17808847-.43280812-.17808847-1.02715396v-3.91481964c0-.59434584.06188374-.80987.17808847-1.02715396.11620474-.21728396.28673046-.38780968.50401441-.50401441.21728396-.11620473.43280813-.17808848 1.02715397-.17808848z" />
                                    </g>
                                </g>
                            </svg>
                        </Frame>
                    </Frame>
                )
                break
            //
            //  ----------------------------------------  iOS 8  ----------------------------------------
            //
            case "iOS8":
                return (
                    <Frame width="100%" height="100%" background="none">
                        {/* Time */}
                        <Frame
                            width={29}
                            height={11}
                            center={"x"}
                            top={4}
                            background="none"
                        >
                            <svg width="29px" height="11px" viewBox="0 0 29 11">
                                <g
                                    stroke="none"
                                    strokeWidth="1"
                                    fill="none"
                                    fillRule="evenodd"
                                >
                                    <g
                                        transform="translate(-190.000000, -329.000000)"
                                        fill={
                                            props.mode === "light"
                                                ? "#000"
                                                : "#FFF"
                                        }
                                        fillRule="nonzero"
                                    >
                                        <path d="M194.157715,339.222168 C196.487305,339.222168 197.871094,337.400391 197.871094,334.309082 C197.871094,331.274902 196.398438,329.611816 194.208496,329.611816 C192.215332,329.611816 190.806152,330.951172 190.806152,332.82373 C190.806152,334.582031 192.075684,335.864258 193.821289,335.864258 C194.894043,335.864258 195.763672,335.369141 196.195312,334.499512 L196.303223,334.499512 C196.28418,336.651367 195.516113,337.882812 194.176758,337.882812 C193.389648,337.882812 192.76123,337.451172 192.564453,336.75293 L190.933105,336.75293 C191.199707,338.244629 192.481934,339.222168 194.157715,339.222168 Z M194.214844,334.601074 C193.148438,334.601074 192.412109,333.852051 192.412109,332.772949 C192.412109,331.738281 193.186523,330.944824 194.221191,330.944824 C195.255859,330.944824 196.030273,331.750977 196.030273,332.804688 C196.030273,333.845703 195.274902,334.601074 194.214844,334.601074 Z M200.486582,333.071289 C201.089609,333.071289 201.502207,332.652344 201.502207,332.087402 C201.502207,331.522461 201.089609,331.103516 200.486582,331.103516 C199.889902,331.103516 199.470957,331.522461 199.470957,332.087402 C199.470957,332.652344 199.889902,333.071289 200.486582,333.071289 Z M200.486582,337.730469 C201.089609,337.730469 201.502207,337.305176 201.502207,336.746582 C201.502207,336.181641 201.089609,335.762695 200.486582,335.762695 C199.889902,335.762695 199.470957,336.181641 199.470957,336.746582 C199.470957,337.305176 199.889902,337.730469 200.486582,337.730469 Z M205.387227,334.912109 L206.536152,334.912109 C207.666035,334.912109 208.351582,335.477051 208.351582,336.397461 C208.351582,337.286133 207.608906,337.882812 206.555195,337.882812 C205.495137,337.882812 204.777852,337.336914 204.708027,336.486328 L203.140156,336.486328 C203.216328,338.143066 204.581074,339.228516 206.574238,339.228516 C208.554707,339.228516 210.027363,338.085938 210.027363,336.511719 C210.027363,335.26123 209.246602,334.455078 208.002461,334.264648 L208.002461,334.156738 C208.973652,333.896484 209.665547,333.166504 209.671895,332.062012 C209.678242,330.748047 208.580098,329.611816 206.631367,329.611816 C204.644551,329.611816 203.40041,330.735352 203.317891,332.341309 L204.854023,332.341309 C204.923848,331.427246 205.545918,330.919434 206.5425,330.919434 C207.513691,330.919434 208.091328,331.516113 208.091328,332.271484 C208.091328,333.103027 207.443867,333.687012 206.504414,333.687012 L205.387227,333.687012 L205.387227,334.912109 Z M214.845488,339.228516 C217.03543,339.228516 218.336699,337.45752 218.336699,334.42334 C218.336699,331.382812 217.016387,329.611816 214.845488,329.611816 C212.668242,329.611816 211.34793,331.382812 211.34793,334.410645 C211.34793,337.451172 212.649199,339.228516 214.845488,339.228516 Z M214.845488,337.901855 C213.683867,337.901855 213.004668,336.695801 213.004668,334.410645 C213.004668,332.144531 213.696562,330.944824 214.845488,330.944824 C215.994414,330.944824 216.679961,332.138184 216.679961,334.410645 C216.679961,336.702148 216.007109,337.901855 214.845488,337.901855 Z"></path>
                                    </g>
                                </g>
                            </svg>
                        </Frame>

                        {/*Cellular*/}
                        <Frame
                            width={17}
                            height={12}
                            top={4}
                            left={5}
                            background="none"
                        >
                            <svg>
                                <path
                                    d="M 1 7.333 L 2 7.333 C 2.552 7.333 3 7.781 3 8.333 L 3 10.333 C 3 10.886 2.552 11.333 2 11.333 L 1 11.333 C 0.448 11.333 0 10.886 0 10.333 L 0 8.333 C 0 7.781 0.448 7.333 1 7.333 Z M 5.667 5.333 L 6.667 5.333 C 7.219 5.333 7.667 5.781 7.667 6.333 L 7.667 10.333 C 7.667 10.886 7.219 11.333 6.667 11.333 L 5.667 11.333 C 5.114 11.333 4.667 10.886 4.667 10.333 L 4.667 6.333 C 4.667 5.781 5.114 5.333 5.667 5.333 Z M 10.333 3 L 11.333 3 C 11.886 3 12.333 3.448 12.333 4 L 12.333 10.333 C 12.333 10.886 11.886 11.333 11.333 11.333 L 10.333 11.333 C 9.781 11.333 9.333 10.886 9.333 10.333 L 9.333 4 C 9.333 3.448 9.781 3 10.333 3 Z M 15 0.667 L 16 0.667 C 16.552 0.667 17 1.114 17 1.667 L 17 10.333 C 17 10.886 16.552 11.333 16 11.333 L 15 11.333 C 14.448 11.333 14 10.886 14 10.333 L 14 1.667 C 14 1.114 14.448 0.667 15 0.667 Z"
                                    fill={
                                        props.mode === "light" ? "#000" : "#FFF"
                                    }
                                />
                            </svg>
                        </Frame>

                        {/* Wifi */}
                        <Frame
                            width={16}
                            height={12}
                            top={4}
                            left={26}
                            background="none"
                        >
                            <svg>
                                <path
                                    d="M 7.997 2.593 C 10.222 2.593 12.361 3.448 13.974 4.979 C 14.095 5.097 14.289 5.096 14.409 4.976 L 15.569 3.805 C 15.63 3.745 15.664 3.662 15.663 3.576 C 15.663 3.49 15.628 3.408 15.567 3.348 C 11.335 -0.704 4.659 -0.704 0.427 3.348 C 0.365 3.408 0.331 3.49 0.33 3.576 C 0.329 3.662 0.363 3.745 0.424 3.805 L 1.585 4.976 C 1.704 5.096 1.898 5.098 2.02 4.979 C 3.632 3.447 5.772 2.593 7.997 2.593 Z M 7.997 6.401 C 9.219 6.401 10.398 6.855 11.304 7.675 C 11.427 7.791 11.62 7.788 11.739 7.669 L 12.899 6.499 C 12.96 6.437 12.993 6.354 12.993 6.267 C 12.992 6.181 12.956 6.098 12.894 6.038 C 10.135 3.474 5.862 3.474 3.103 6.038 C 3.04 6.098 3.005 6.181 3.004 6.267 C 3.003 6.354 3.037 6.437 3.098 6.499 L 4.257 7.669 C 4.377 7.788 4.57 7.791 4.692 7.675 C 5.598 6.855 6.776 6.401 7.997 6.401 Z M 10.225 9.194 C 10.287 9.133 10.321 9.05 10.319 8.963 C 10.318 8.876 10.28 8.794 10.216 8.736 C 8.935 7.653 7.059 7.653 5.779 8.736 C 5.714 8.794 5.677 8.876 5.675 8.963 C 5.673 9.049 5.707 9.133 5.769 9.194 L 7.775 11.216 C 7.833 11.275 7.913 11.309 7.997 11.309 C 8.081 11.309 8.161 11.275 8.22 11.216 Z"
                                    fill={
                                        props.mode === "light" ? "#000" : "#FFF"
                                    }
                                />
                            </svg>
                        </Frame>

                        {/* Battery */}
                        <Frame
                            width={25}
                            height={12}
                            top={4}
                            right={5}
                            background="none"
                        >
                            <svg>
                                <g fill="none" fillRule="evenodd">
                                    <path
                                        d="m3.7518137.83333333c-1.00827318 0-1.40697456.07699279-1.8185085.29708355-.34743368.18580957-.61711209.45548797-.80292165.80292165-.22009076.41153394-.29708355.81023533-.29708355 1.81850851v4.49630592c0 1.00827318.07699279 1.40697457.29708355 1.81850854.18580956.3474336.45548797.6171121.80292165.8029216.41153394.2200908.81023532.2970836 1.8185085.2970836h15.1629726c1.0082732 0 1.4069746-.0769928 1.8185085-.2970836.3474337-.1858095.6171121-.455488.8029217-.8029216.2200907-.41153397.2970835-.81023536.2970835-1.81850854v-4.49630592c0-1.00827318-.0769928-1.40697457-.2970835-1.81850851-.1858096-.34743368-.455488-.61711208-.8029217-.80292165-.4115339-.22009076-.8102353-.29708355-1.8185085-.29708355z"
                                        opacity=".35"
                                        stroke={
                                            props.mode === "light"
                                                ? "#000"
                                                : "#FFF"
                                        }
                                    />
                                    <g
                                        fill={
                                            props.mode === "light"
                                                ? "#000"
                                                : "#FFF"
                                        }
                                        fillRule="nonzero"
                                    >
                                        <path
                                            d="m23.3333 4v4c.8047311-.33877652 1.328038-1.12686672 1.328038-2s-.5233069-1.66122348-1.328038-2"
                                            opacity=".4"
                                        />
                                        <path d="m4.04255685 2.33333333h14.58148625c.5943459 0 .8098701.06188375 1.027154.17808848.217284.11620473.3878097.28673045.5040144.50401441.1162048.21728396.1780885.43280812.1780885 1.02715396v3.91481964c0 .59434584-.0618837.80987-.1780885 1.02715396-.1162047.21728396-.2867304.38780968-.5040144.50401441-.2172839.11620473-.4328081.17808848-1.027154.17808848h-14.58148625c-.59434584 0-.80987001-.06188375-1.02715397-.17808848-.21728395-.11620473-.38780967-.28673045-.50401441-.50401441-.11620473-.21728396-.17808847-.43280812-.17808847-1.02715396v-3.91481964c0-.59434584.06188374-.80987.17808847-1.02715396.11620474-.21728396.28673046-.38780968.50401441-.50401441.21728396-.11620473.43280813-.17808848 1.02715397-.17808848z" />
                                    </g>
                                </g>
                            </svg>
                        </Frame>
                    </Frame>
                )
                break
            //
            //  ----------------------------------------  iOS X  ----------------------------------------
            //
            case "pixel3":
                return (
                    <Frame width="100%" height="100%" background="none">
                        {/* Time */}
                        <Frame
                            name="Time"
                            width={27}
                            height={12}
                            top={6}
                            left={17}
                            background="none"
                        >
                            <svg width="27px" height="12px" viewBox="0 0 27 12">
                                <g
                                    stroke="none"
                                    strokeWidth="1"
                                    fill="none"
                                    fillRule="evenodd"
                                >
                                    <path
                                        d="M5.45507812,6.83691406 C4.89908576,7.42480763 4.24967819,7.71875 3.50683594,7.71875 C2.63183156,7.71875 1.93457291,7.4122752 1.41503906,6.79931641 C0.895505215,6.18635761 0.635742188,5.38086437 0.635742188,4.3828125 C0.635742188,3.72655922 0.767902324,3.1318386 1.03222656,2.59863281 C1.2965508,2.06542702 1.6725236,1.65071763 2.16015625,1.35449219 C2.6477889,1.05826675 3.21288741,0.91015625 3.85546875,0.91015625 C4.85807793,0.91015625 5.65331737,1.28385043 6.24121094,2.03125 C6.8291045,2.77864957 7.12304688,3.77896509 7.12304688,5.03222656 L7.12304688,5.49707031 C7.12304688,7.29265221 6.71745197,8.66210466 5.90625,9.60546875 C5.09504803,10.5488328 3.88509919,11.0273437 2.27636719,11.0410156 L2.078125,11.0410156 L2.078125,9.66699219 L2.31738281,9.66699219 C3.29720542,9.65332024 4.04915103,9.41748276 4.57324219,8.95947266 C5.09733335,8.50146255 5.39127572,7.7939501 5.45507812,6.83691406 Z M3.85546875,6.41992188 C4.18815271,6.41992188 4.50032406,6.32421971 4.79199219,6.1328125 C5.08366031,5.94140529 5.30924399,5.67708502 5.46875,5.33984375 L5.46875,4.69042969 C5.46875,3.95670206 5.31836088,3.36653869 5.01757812,2.91992188 C4.71679537,2.47330506 4.32715083,2.25 3.84863281,2.25 C3.37011479,2.25 2.98844543,2.44710089 2.70361328,2.84130859 C2.41878113,3.23551629 2.27636719,3.73339543 2.27636719,4.33496094 C2.27636719,4.95931302 2.41992044,5.46288871 2.70703125,5.84570312 C2.99414206,6.22851754 3.37695073,6.41992188 3.85546875,6.41992188 Z M9.80957031,9.25 C10.1194677,9.25 10.3587231,9.33772699 10.5273438,9.51318359 C10.6959644,9.6886402 10.7802734,9.90852732 10.7802734,10.1728516 C10.7802734,10.4326185 10.6959644,10.6479484 10.5273438,10.8188477 C10.3587231,10.9897469 10.1194677,11.0751953 9.80957031,11.0751953 C9.51334487,11.0751953 9.2786467,10.9908863 9.10546875,10.8222656 C8.9322908,10.653645 8.84570312,10.4371758 8.84570312,10.1728516 C8.84570312,9.90852732 8.93001218,9.6886402 9.09863281,9.51318359 C9.26725345,9.33772699 9.50423024,9.25 9.80957031,9.25 Z M9.80957031,3.44628906 C10.1194677,3.44628906 10.3587231,3.53401605 10.5273438,3.70947266 C10.6959644,3.88492926 10.7802734,4.10481639 10.7802734,4.36914062 C10.7802734,4.62890755 10.6959644,4.84423743 10.5273438,5.01513672 C10.3587231,5.18603601 10.1194677,5.27148438 9.80957031,5.27148438 C9.51334487,5.27148438 9.2786467,5.18717532 9.10546875,5.01855469 C8.9322908,4.84993405 8.84570312,4.63346486 8.84570312,4.36914062 C8.84570312,4.10481639 8.93001218,3.88492926 9.09863281,3.70947266 C9.26725345,3.53401605 9.50423024,3.44628906 9.80957031,3.44628906 Z M14.3349609,5.26464844 L15.3466797,5.26464844 C15.8570989,5.26009112 16.2615545,5.12793099 16.5600586,4.86816406 C16.8585627,4.60839714 17.0078125,4.23242434 17.0078125,3.74023438 C17.0078125,3.26627367 16.8836275,2.89827605 16.6352539,2.63623047 C16.3868803,2.37418489 16.0074895,2.24316406 15.4970703,2.24316406 C15.0504535,2.24316406 14.6835952,2.37190626 14.3964844,2.62939453 C14.1093736,2.8868828 13.9658203,3.2229797 13.9658203,3.63769531 L12.3046875,3.63769531 C12.3046875,3.12727609 12.4402656,2.66243699 12.7114258,2.24316406 C12.982586,1.82389113 13.3608374,1.49690873 13.8461914,1.26220703 C14.3315454,1.02750534 14.874997,0.91015625 15.4765625,0.91015625 C16.4700571,0.91015625 17.2504854,1.15966547 17.8178711,1.65869141 C18.3852567,2.15771734 18.6689453,2.85155806 18.6689453,3.74023438 C18.6689453,4.18685119 18.5265314,4.60725714 18.2416992,5.00146484 C17.9568671,5.39567254 17.5888694,5.69303285 17.1376953,5.89355469 C17.684573,6.08040458 18.1004218,6.37434695 18.3852539,6.77539062 C18.6700861,7.1764343 18.8125,7.65494514 18.8125,8.2109375 C18.8125,9.10417113 18.5060252,9.81510152 17.8930664,10.34375 C17.2801076,10.8723985 16.4746144,11.1367188 15.4765625,11.1367188 C14.5195265,11.1367188 13.7356801,10.881513 13.125,10.3710938 C12.5143199,9.86067453 12.2089844,9.18164486 12.2089844,8.33398438 L13.8701172,8.33398438 C13.8701172,8.77148656 14.0170884,9.12695176 14.3110352,9.40039062 C14.6049819,9.67382949 15.000323,9.81054688 15.4970703,9.81054688 C16.0120468,9.81054688 16.4176417,9.67382949 16.7138672,9.40039062 C17.0100926,9.12695176 17.1582031,8.73047135 17.1582031,8.2109375 C17.1582031,7.68684634 17.0032568,7.28353006 16.6933594,7.00097656 C16.383462,6.71842307 15.9231801,6.57714844 15.3125,6.57714844 L14.3349609,6.57714844 L14.3349609,5.26464844 Z M23.5976562,0.91015625 C24.6868544,0.91015625 25.4991835,1.25878558 26.034668,1.95605469 C26.5701524,2.6533238 26.8447265,3.69693315 26.8583984,5.08691406 L26.8583984,6.80957031 C26.8583984,8.24056705 26.5906602,9.31949506 26.0551758,10.0463867 C25.5196913,10.7732784 24.7050836,11.1367188 23.6113281,11.1367188 C22.5358019,11.1367188 21.7257514,10.7823929 21.1811523,10.0737305 C20.6365533,9.36506807 20.3574219,8.31576216 20.34375,6.92578125 L20.34375,5.203125 C20.34375,3.77212826 20.6114882,2.69889681 21.1469727,1.98339844 C21.6824571,1.26790007 22.4993435,0.91015625 23.5976562,0.91015625 Z M23.5976562,2.24316406 C23.0553358,2.24316406 22.6565768,2.44937945 22.4013672,2.86181641 C22.1461576,3.27425336 22.0117188,3.91796437 21.9980469,4.79296875 L21.9980469,4.79296875 L21.9980469,7.04882812 C21.9980469,7.97852027 22.1279284,8.67122168 22.3876953,9.12695312 C22.6474622,9.58268457 23.0553358,9.81054688 23.6113281,9.81054688 C24.1445339,9.81054688 24.5387357,9.59977424 24.7939453,9.17822266 C25.0491549,8.75667107 25.1835937,8.0970097 25.1972656,7.19921875 L25.1972656,7.19921875 L25.1972656,4.95019531 C25.1972656,4.01594585 25.0696627,3.33121962 24.8144531,2.89599609 C24.5592435,2.46077256 24.1536486,2.24316406 23.5976562,2.24316406 Z"
                                        fill={
                                            props.mode === "light"
                                                ? "#777"
                                                : "#FFF"
                                        }
                                        fillRule="nonzero"
                                    ></path>
                                </g>
                            </svg>
                        </Frame>

                        {/*Wifi*/}
                        <Frame
                            width={12}
                            height={16}
                            top={6}
                            right={49}
                            background="none"
                        >
                            <svg height="12" viewBox="0 0 16 12" width="16">
                                <path
                                    d="m349.5 9.5c2.5-2.33333333 5-3.5 7.5-3.5s5 1.16666667 7.5 3.5l-7.5 8.5z"
                                    fill={
                                        props.mode === "light" ? "#777" : "#FFF"
                                    }
                                    fillRule="evenodd"
                                    transform="translate(-349 -6)"
                                />
                            </svg>
                        </Frame>

                        {/* Cellular */}
                        <Frame
                            width={12}
                            height={12}
                            top={6}
                            right={32}
                            background="none"
                        >
                            <svg height="12" viewBox="0 0 12 12" width="12">
                                <path
                                    d="m367 18 12-12v12z"
                                    fill={
                                        props.mode === "light" ? "#777" : "#FFF"
                                    }
                                    fillRule="evenodd"
                                    transform="translate(-367 -6)"
                                />
                            </svg>
                        </Frame>

                        {/* Battery */}
                        <Frame
                            width={8}
                            height={12}
                            top={6}
                            right={17}
                            background="none"
                        >
                            <svg height="12" viewBox="0 0 8 12" width="8">
                                <path
                                    d="m388 8v-2h4v2h2v10h-8v-10z"
                                    fill={
                                        props.mode === "light" ? "#777" : "#FFF"
                                    }
                                    fillRule="evenodd"
                                    transform="translate(-386 -6)"
                                />
                            </svg>
                        </Frame>
                    </Frame>
                )
                break
        }
    }

    const statusBarContent = React.useMemo(() => renderStatusBarContent(), [
        props,
    ])

    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                   HEIGHT                                                      */
    /* ------------------------------------------------------------------------------------------------------------- */

    const getStatusBarHeight = () => {
        switch (props.type) {
            case "iOSX":
                return 44
                break
            case "iOS8":
                return 20
                break
            case "pixel3":
                return 25
                break
        }
    }

    const statusBarHeight = React.useMemo(() => getStatusBarHeight(), [props])

    /* ------------------------------------------------------------------------------------------------------------- */
    /*                                                    MAIN                                                       */
    /* ------------------------------------------------------------------------------------------------------------- */

    return (
        <Frame
            width="100%"
            height={statusBarHeight}
            background="none"
            overflow="hidden"
            style={{
                zIndex: 1000,
            }}
        >
            {statusBarContent}
        </Frame>
    )
}

/* ------------------------------------------------------------------------------------------------------------- */
/*                                                   PROPERTIES                                                  */
/* ------------------------------------------------------------------------------------------------------------- */

Status_Bar.defaultProps = {
    width: 375,
    height: 44,
    type: "iOSX",
    mode: "light",
}

addPropertyControls(Status_Bar, {
    type: {
        title: "Type",
        type: ControlType.Enum,
        options: ["iOSX", "iOS8", "pixel3"],
        optionTitles: ["iOS X (44)", "iOS 8 (20)", "Pixel 3 (25)"],
    },
    mode: {
        title: "Mode",
        type: ControlType.SegmentedEnum,
        options: ["light", "dark"],
        optionTitles: ["Light", "Dark"],
    },
})
